{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "interpreted-generic",
   "metadata": {},
   "outputs": [],
   "source": [
    "#default_exp phenotypes.__init__"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fluid-birmingham",
   "metadata": {},
   "source": [
    "# Functions to extract phenotype info\n",
    "\n",
    "> API details."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "psychological-policy",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "from combinatorial_gwas.data_catalog import get_catalog, get_parameters\n",
    "import combinatorial_gwas\n",
    "from pathlib import Path\n",
    "import pandas as pd\n",
    "from dataclasses import dataclass\n",
    "from functools import partial\n",
    "import numpy as np\n",
    "from typing import List, Union\n",
    "from fastcore.utils import partialler\n",
    "import logging"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fitted-marsh",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-21-dcb25e435725>:5: UserWarning: registration of accessor <class '__main__.QueryDataframe'> under name 'pheno' for type <class 'pandas.core.frame.DataFrame'> is overriding a preexisting attribute with the same name.\n",
      "  class QueryDataframe():\n"
     ]
    }
   ],
   "source": [
    "#export\n",
    "\n",
    "@pd.api.extensions.register_dataframe_accessor(\"pheno\")\n",
    "@dataclass\n",
    "class QueryDataframe():\n",
    "    df: pd.DataFrame\n",
    "    \n",
    "    def query(self, **column_dict:dict):\n",
    "        query_str = \" and \".join([f\"({col} {cond})\" for col, cond in column_dict.items()])\n",
    "        return self.df.query(query_str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "whole-alignment",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'example_test_data_ratio': 0.2,\n",
       " 'example_num_train_iter': 10000,\n",
       " 'example_learning_rate': 0.01,\n",
       " 'template_gwas_result_file_link': 'https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/{phenotype_code}.gwas.imputed_v3.both_sexes.tsv.bgz',\n",
       " 'genetic_file_path_template': '/lab/corradin_biobank/Raw_UKB_downloads/BGEN/ukb_imp_chr{chrom_number}_v3.bgen',\n",
       " 'sample_file_template': '/lab/corradin_biobank/Raw_UKB_downloads/sample_files/ukb45624_imp_chr21_v3_s487275.sample',\n",
       " 'data_dict_file': '/lab/corradin_biobank/Phenotypes//Data_Dictionary_Showcase.csv',\n",
       " 'pheno_file': '/lab/corradin_biobank/samples//neale_gwas_both_sexes_parquet/',\n",
       " 'coding_file_path_template': '/lab/corradin_biobank/Phenotypes//data_codes/datacode-*.tsv',\n",
       " 'id_col': 'f.eid'}"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#export\n",
    "parameters = get_parameters()\n",
    "parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "authentic-miami",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['example_iris_data',\n",
       " 'globals_test',\n",
       " 'disease_summary',\n",
       " 'ICD10_pheno_matrix',\n",
       " 'phenotypes_info_Neale',\n",
       " 'GWAS_results_links_df',\n",
       " 'heritability_trait_level_summary',\n",
       " 'heritability_trait_level_partitioned_z4_sig']"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#export\n",
    "catalog_all = get_catalog()\n",
    "catalog_all = catalog_all.reload()\n",
    "catalog_all.list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "advised-printer",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "read_csv_compressed= partialler(pd.read_csv, sep=\"\\t\", compression= \"gzip\")\n",
    "get_GWAS_result_link = partialler(parameters['template_gwas_result_file_link'].format)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "introductory-pakistan",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "political-pennsylvania",
   "metadata": {},
   "source": [
    "### Find high heritability traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "centered-crowd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/extras/datasets/pandas/csv_dataset.py:158: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return pd.read_csv(fs_file, **self._load_args)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phenotype</th>\n",
       "      <th>description</th>\n",
       "      <th>h2_liability</th>\n",
       "      <th>h2_liability_se</th>\n",
       "      <th>h2_observed</th>\n",
       "      <th>h2_observed_se</th>\n",
       "      <th>h2_z</th>\n",
       "      <th>h2_p</th>\n",
       "      <th>h2_sig</th>\n",
       "      <th>confidence</th>\n",
       "      <th>...</th>\n",
       "      <th>sex</th>\n",
       "      <th>isNotPrimary</th>\n",
       "      <th>isBadPower</th>\n",
       "      <th>isLowNeff</th>\n",
       "      <th>isMidNeff</th>\n",
       "      <th>isExtremeSE</th>\n",
       "      <th>isHighSE</th>\n",
       "      <th>isSexBias</th>\n",
       "      <th>isBadOrdinal</th>\n",
       "      <th>isNumericOrdinal</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100001_irnt</td>\n",
       "      <td>Food weight</td>\n",
       "      <td>0.068818</td>\n",
       "      <td>0.016857</td>\n",
       "      <td>0.068818</td>\n",
       "      <td>0.016857</td>\n",
       "      <td>4.082528</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>...</td>\n",
       "      <td>both_sexes</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100002_irnt</td>\n",
       "      <td>Energy</td>\n",
       "      <td>0.064783</td>\n",
       "      <td>0.015784</td>\n",
       "      <td>0.064783</td>\n",
       "      <td>0.015784</td>\n",
       "      <td>4.104249</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>...</td>\n",
       "      <td>both_sexes</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100003_irnt</td>\n",
       "      <td>Protein</td>\n",
       "      <td>0.034543</td>\n",
       "      <td>0.014619</td>\n",
       "      <td>0.034543</td>\n",
       "      <td>0.014619</td>\n",
       "      <td>2.362892</td>\n",
       "      <td>0.009066</td>\n",
       "      <td>nominal</td>\n",
       "      <td>high</td>\n",
       "      <td>...</td>\n",
       "      <td>both_sexes</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100004_irnt</td>\n",
       "      <td>Fat</td>\n",
       "      <td>0.055197</td>\n",
       "      <td>0.018265</td>\n",
       "      <td>0.055197</td>\n",
       "      <td>0.018265</td>\n",
       "      <td>3.021937</td>\n",
       "      <td>0.001256</td>\n",
       "      <td>nominal</td>\n",
       "      <td>high</td>\n",
       "      <td>...</td>\n",
       "      <td>both_sexes</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100005_irnt</td>\n",
       "      <td>Carbohydrate</td>\n",
       "      <td>0.051744</td>\n",
       "      <td>0.015357</td>\n",
       "      <td>0.051744</td>\n",
       "      <td>0.015357</td>\n",
       "      <td>3.369359</td>\n",
       "      <td>0.000377</td>\n",
       "      <td>nominal</td>\n",
       "      <td>high</td>\n",
       "      <td>...</td>\n",
       "      <td>both_sexes</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     phenotype   description  h2_liability  h2_liability_se  h2_observed  \\\n",
       "0  100001_irnt   Food weight      0.068818         0.016857     0.068818   \n",
       "1  100002_irnt        Energy      0.064783         0.015784     0.064783   \n",
       "2  100003_irnt       Protein      0.034543         0.014619     0.034543   \n",
       "3  100004_irnt           Fat      0.055197         0.018265     0.055197   \n",
       "4  100005_irnt  Carbohydrate      0.051744         0.015357     0.051744   \n",
       "\n",
       "   h2_observed_se      h2_z      h2_p   h2_sig confidence  ...         sex  \\\n",
       "0        0.016857  4.082528  0.000022       z4       high  ...  both_sexes   \n",
       "1        0.015784  4.104249  0.000020       z4       high  ...  both_sexes   \n",
       "2        0.014619  2.362892  0.009066  nominal       high  ...  both_sexes   \n",
       "3        0.018265  3.021937  0.001256  nominal       high  ...  both_sexes   \n",
       "4        0.015357  3.369359  0.000377  nominal       high  ...  both_sexes   \n",
       "\n",
       "   isNotPrimary  isBadPower  isLowNeff  isMidNeff  isExtremeSE  isHighSE  \\\n",
       "0         False       False      False      False        False     False   \n",
       "1         False       False      False      False        False     False   \n",
       "2         False       False      False      False        False     False   \n",
       "3         False       False      False      False        False     False   \n",
       "4         False       False      False      False        False     False   \n",
       "\n",
       "   isSexBias  isBadOrdinal  isNumericOrdinal  \n",
       "0      False         False             False  \n",
       "1      False         False             False  \n",
       "2      False         False             False  \n",
       "3      False         False             False  \n",
       "4      False         False             False  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "heritability_Neale = catalog_all.load(\"heritability_trait_level_summary\")\n",
    "heritability_Neale.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "middle-jungle",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['phenotype', 'description', 'h2_liability', 'h2_liability_se',\n",
       "       'h2_observed', 'h2_observed_se', 'h2_z', 'h2_p', 'h2_sig', 'confidence',\n",
       "       'notes', 'intercept', 'intercept_se', 'intercept_z', 'intercept_p',\n",
       "       'lambdaGC', 'mean_chi2', 'ratio', 'ratio_se', 'n', 'Neff',\n",
       "       'variable_type', 'isBinary', 'n_cases', 'n_controls', 'prevalence',\n",
       "       'source', 'sex', 'isNotPrimary', 'isBadPower', 'isLowNeff', 'isMidNeff',\n",
       "       'isExtremeSE', 'isHighSE', 'isSexBias', 'isBadOrdinal',\n",
       "       'isNumericOrdinal'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "heritability_Neale.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "informative-wrong",
   "metadata": {},
   "outputs": [],
   "source": [
    "display_cols = ['description', 'h2_liability', 'h2_sig', 'confidence', 'n_cases', 'n_controls', 'prevalence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "rolled-iceland",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>h2_liability</th>\n",
       "      <th>h2_sig</th>\n",
       "      <th>confidence</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>n_controls</th>\n",
       "      <th>prevalence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>I83</th>\n",
       "      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>\n",
       "      <td>0.189299</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>8763.0</td>\n",
       "      <td>352431.0</td>\n",
       "      <td>0.024261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I25</th>\n",
       "      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>\n",
       "      <td>0.163823</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>12769.0</td>\n",
       "      <td>348425.0</td>\n",
       "      <td>0.035352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G56</th>\n",
       "      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>\n",
       "      <td>0.157519</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>8130.0</td>\n",
       "      <td>353064.0</td>\n",
       "      <td>0.022509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N81</th>\n",
       "      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>\n",
       "      <td>0.152223</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>7511.0</td>\n",
       "      <td>186663.0</td>\n",
       "      <td>0.038682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M16</th>\n",
       "      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>\n",
       "      <td>0.151297</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>9136.0</td>\n",
       "      <td>352058.0</td>\n",
       "      <td>0.025294</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 description  h2_liability  \\\n",
       "phenotype                                                                    \n",
       "I83        Diagnoses - main ICD10: I83 Varicose veins of ...      0.189299   \n",
       "I25        Diagnoses - main ICD10: I25 Chronic ischaemic ...      0.163823   \n",
       "G56        Diagnoses - main ICD10: G56 Mononeuropathies o...      0.157519   \n",
       "N81        Diagnoses - main ICD10: N81 Female genital pro...      0.152223   \n",
       "M16        Diagnoses - main ICD10: M16 Coxarthrosis [arth...      0.151297   \n",
       "\n",
       "          h2_sig confidence  n_cases  n_controls  prevalence  \n",
       "phenotype                                                     \n",
       "I83           z7     medium   8763.0    352431.0    0.024261  \n",
       "I25           z7     medium  12769.0    348425.0    0.035352  \n",
       "G56           z7     medium   8130.0    353064.0    0.022509  \n",
       "N81           z4     medium   7511.0    186663.0    0.038682  \n",
       "M16           z4     medium   9136.0    352058.0    0.025294  "
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "quality_heritability_phenos = heritability_Neale.pheno.query(h2_sig = \"in ['z7', 'z4']\", source= \" == 'icd10'\", confidence= \"in ['medium', 'high']\").sort_values(\"h2_liability\", ascending = False)\n",
    "quality_heritability_phenos = quality_heritability_phenos.set_index(\"phenotype\")\n",
    "quality_heritability_phenos.head()[display_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fitting-aspect",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "      <th>h2_liability</th>\n",
       "      <th>h2_sig</th>\n",
       "      <th>confidence</th>\n",
       "      <th>n_cases</th>\n",
       "      <th>n_controls</th>\n",
       "      <th>prevalence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>I83</th>\n",
       "      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>\n",
       "      <td>0.189299</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>8763.0</td>\n",
       "      <td>352431.0</td>\n",
       "      <td>0.024261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I25</th>\n",
       "      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>\n",
       "      <td>0.163823</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>12769.0</td>\n",
       "      <td>348425.0</td>\n",
       "      <td>0.035352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G56</th>\n",
       "      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>\n",
       "      <td>0.157519</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>8130.0</td>\n",
       "      <td>353064.0</td>\n",
       "      <td>0.022509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N81</th>\n",
       "      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>\n",
       "      <td>0.152223</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>7511.0</td>\n",
       "      <td>186663.0</td>\n",
       "      <td>0.038682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M16</th>\n",
       "      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>\n",
       "      <td>0.151297</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>9136.0</td>\n",
       "      <td>352058.0</td>\n",
       "      <td>0.025294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M17</th>\n",
       "      <td>Diagnoses - main ICD10: M17 Gonarthrosis [arth...</td>\n",
       "      <td>0.144360</td>\n",
       "      <td>z7</td>\n",
       "      <td>high</td>\n",
       "      <td>11497.0</td>\n",
       "      <td>349697.0</td>\n",
       "      <td>0.031831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I48</th>\n",
       "      <td>Diagnoses - main ICD10: I48 Atrial fibrillatio...</td>\n",
       "      <td>0.144326</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>6356.0</td>\n",
       "      <td>354838.0</td>\n",
       "      <td>0.017597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C44</th>\n",
       "      <td>Diagnoses - main ICD10: C44 Other malignant ne...</td>\n",
       "      <td>0.141495</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>9086.0</td>\n",
       "      <td>352108.0</td>\n",
       "      <td>0.025155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K80</th>\n",
       "      <td>Diagnoses - main ICD10: K80 Cholelithiasis</td>\n",
       "      <td>0.137236</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>10520.0</td>\n",
       "      <td>350674.0</td>\n",
       "      <td>0.029126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I21</th>\n",
       "      <td>Diagnoses - main ICD10: I21 Acute myocardial i...</td>\n",
       "      <td>0.136628</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>5948.0</td>\n",
       "      <td>355246.0</td>\n",
       "      <td>0.016468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I20</th>\n",
       "      <td>Diagnoses - main ICD10: I20 Angina pectoris</td>\n",
       "      <td>0.129187</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>6246.0</td>\n",
       "      <td>354948.0</td>\n",
       "      <td>0.017293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K57</th>\n",
       "      <td>Diagnoses - main ICD10: K57 Diverticular disea...</td>\n",
       "      <td>0.129184</td>\n",
       "      <td>z7</td>\n",
       "      <td>high</td>\n",
       "      <td>12662.0</td>\n",
       "      <td>348532.0</td>\n",
       "      <td>0.035056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K40</th>\n",
       "      <td>Diagnoses - main ICD10: K40 Inguinal hernia</td>\n",
       "      <td>0.120213</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>13147.0</td>\n",
       "      <td>348047.0</td>\n",
       "      <td>0.036399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>D12</th>\n",
       "      <td>Diagnoses - main ICD10: D12 Benign neoplasm of...</td>\n",
       "      <td>0.118093</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>8877.0</td>\n",
       "      <td>352317.0</td>\n",
       "      <td>0.024577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M54</th>\n",
       "      <td>Diagnoses - main ICD10: M54 Dorsalgia</td>\n",
       "      <td>0.114167</td>\n",
       "      <td>z7</td>\n",
       "      <td>medium</td>\n",
       "      <td>8361.0</td>\n",
       "      <td>352833.0</td>\n",
       "      <td>0.023148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C50</th>\n",
       "      <td>Diagnoses - main ICD10: C50 Malignant neoplasm...</td>\n",
       "      <td>0.110280</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>8304.0</td>\n",
       "      <td>352890.0</td>\n",
       "      <td>0.022990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K22</th>\n",
       "      <td>Diagnoses - main ICD10: K22 Other diseases of ...</td>\n",
       "      <td>0.107578</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>5494.0</td>\n",
       "      <td>355700.0</td>\n",
       "      <td>0.015211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R07</th>\n",
       "      <td>Diagnoses - main ICD10: R07 Pain in throat and...</td>\n",
       "      <td>0.100279</td>\n",
       "      <td>z7</td>\n",
       "      <td>high</td>\n",
       "      <td>24530.0</td>\n",
       "      <td>336664.0</td>\n",
       "      <td>0.067914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M20</th>\n",
       "      <td>Diagnoses - main ICD10: M20 Acquired deformiti...</td>\n",
       "      <td>0.096417</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>7773.0</td>\n",
       "      <td>353421.0</td>\n",
       "      <td>0.021520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H26</th>\n",
       "      <td>Diagnoses - main ICD10: H26 Other cataract</td>\n",
       "      <td>0.086648</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>11306.0</td>\n",
       "      <td>349888.0</td>\n",
       "      <td>0.031302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K21</th>\n",
       "      <td>Diagnoses - main ICD10: K21 Gastro-oesophageal...</td>\n",
       "      <td>0.081990</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>10743.0</td>\n",
       "      <td>350451.0</td>\n",
       "      <td>0.029743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>K44</th>\n",
       "      <td>Diagnoses - main ICD10: K44 Diaphragmatic hernia</td>\n",
       "      <td>0.075533</td>\n",
       "      <td>z4</td>\n",
       "      <td>medium</td>\n",
       "      <td>8042.0</td>\n",
       "      <td>353152.0</td>\n",
       "      <td>0.022265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>M23</th>\n",
       "      <td>Diagnoses - main ICD10: M23 Internal derangeme...</td>\n",
       "      <td>0.074823</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>11831.0</td>\n",
       "      <td>349363.0</td>\n",
       "      <td>0.032755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>N39</th>\n",
       "      <td>Diagnoses - main ICD10: N39 Other disorders of...</td>\n",
       "      <td>0.063935</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>10551.0</td>\n",
       "      <td>350643.0</td>\n",
       "      <td>0.029211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R10</th>\n",
       "      <td>Diagnoses - main ICD10: R10 Abdominal and pelv...</td>\n",
       "      <td>0.062420</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>20240.0</td>\n",
       "      <td>340954.0</td>\n",
       "      <td>0.056036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>R31</th>\n",
       "      <td>Diagnoses - main ICD10: R31 Unspecified haemat...</td>\n",
       "      <td>0.060724</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>11283.0</td>\n",
       "      <td>349911.0</td>\n",
       "      <td>0.031238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>I84</th>\n",
       "      <td>Diagnoses - main ICD10: I84 Haemorrhoids</td>\n",
       "      <td>0.057475</td>\n",
       "      <td>z4</td>\n",
       "      <td>high</td>\n",
       "      <td>12102.0</td>\n",
       "      <td>349092.0</td>\n",
       "      <td>0.033506</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 description  h2_liability  \\\n",
       "phenotype                                                                    \n",
       "I83        Diagnoses - main ICD10: I83 Varicose veins of ...      0.189299   \n",
       "I25        Diagnoses - main ICD10: I25 Chronic ischaemic ...      0.163823   \n",
       "G56        Diagnoses - main ICD10: G56 Mononeuropathies o...      0.157519   \n",
       "N81        Diagnoses - main ICD10: N81 Female genital pro...      0.152223   \n",
       "M16        Diagnoses - main ICD10: M16 Coxarthrosis [arth...      0.151297   \n",
       "M17        Diagnoses - main ICD10: M17 Gonarthrosis [arth...      0.144360   \n",
       "I48        Diagnoses - main ICD10: I48 Atrial fibrillatio...      0.144326   \n",
       "C44        Diagnoses - main ICD10: C44 Other malignant ne...      0.141495   \n",
       "K80               Diagnoses - main ICD10: K80 Cholelithiasis      0.137236   \n",
       "I21        Diagnoses - main ICD10: I21 Acute myocardial i...      0.136628   \n",
       "I20              Diagnoses - main ICD10: I20 Angina pectoris      0.129187   \n",
       "K57        Diagnoses - main ICD10: K57 Diverticular disea...      0.129184   \n",
       "K40              Diagnoses - main ICD10: K40 Inguinal hernia      0.120213   \n",
       "D12        Diagnoses - main ICD10: D12 Benign neoplasm of...      0.118093   \n",
       "M54                    Diagnoses - main ICD10: M54 Dorsalgia      0.114167   \n",
       "C50        Diagnoses - main ICD10: C50 Malignant neoplasm...      0.110280   \n",
       "K22        Diagnoses - main ICD10: K22 Other diseases of ...      0.107578   \n",
       "R07        Diagnoses - main ICD10: R07 Pain in throat and...      0.100279   \n",
       "M20        Diagnoses - main ICD10: M20 Acquired deformiti...      0.096417   \n",
       "H26               Diagnoses - main ICD10: H26 Other cataract      0.086648   \n",
       "K21        Diagnoses - main ICD10: K21 Gastro-oesophageal...      0.081990   \n",
       "K44         Diagnoses - main ICD10: K44 Diaphragmatic hernia      0.075533   \n",
       "M23        Diagnoses - main ICD10: M23 Internal derangeme...      0.074823   \n",
       "N39        Diagnoses - main ICD10: N39 Other disorders of...      0.063935   \n",
       "R10        Diagnoses - main ICD10: R10 Abdominal and pelv...      0.062420   \n",
       "R31        Diagnoses - main ICD10: R31 Unspecified haemat...      0.060724   \n",
       "I84                 Diagnoses - main ICD10: I84 Haemorrhoids      0.057475   \n",
       "\n",
       "          h2_sig confidence  n_cases  n_controls  prevalence  \n",
       "phenotype                                                     \n",
       "I83           z7     medium   8763.0    352431.0    0.024261  \n",
       "I25           z7     medium  12769.0    348425.0    0.035352  \n",
       "G56           z7     medium   8130.0    353064.0    0.022509  \n",
       "N81           z4     medium   7511.0    186663.0    0.038682  \n",
       "M16           z4     medium   9136.0    352058.0    0.025294  \n",
       "M17           z7       high  11497.0    349697.0    0.031831  \n",
       "I48           z4     medium   6356.0    354838.0    0.017597  \n",
       "C44           z4     medium   9086.0    352108.0    0.025155  \n",
       "K80           z4       high  10520.0    350674.0    0.029126  \n",
       "I21           z4     medium   5948.0    355246.0    0.016468  \n",
       "I20           z4     medium   6246.0    354948.0    0.017293  \n",
       "K57           z7       high  12662.0    348532.0    0.035056  \n",
       "K40           z7     medium  13147.0    348047.0    0.036399  \n",
       "D12           z4     medium   8877.0    352317.0    0.024577  \n",
       "M54           z7     medium   8361.0    352833.0    0.023148  \n",
       "C50           z4     medium   8304.0    352890.0    0.022990  \n",
       "K22           z4     medium   5494.0    355700.0    0.015211  \n",
       "R07           z7       high  24530.0    336664.0    0.067914  \n",
       "M20           z4     medium   7773.0    353421.0    0.021520  \n",
       "H26           z4       high  11306.0    349888.0    0.031302  \n",
       "K21           z4       high  10743.0    350451.0    0.029743  \n",
       "K44           z4     medium   8042.0    353152.0    0.022265  \n",
       "M23           z4       high  11831.0    349363.0    0.032755  \n",
       "N39           z4       high  10551.0    350643.0    0.029211  \n",
       "R10           z4       high  20240.0    340954.0    0.056036  \n",
       "R31           z4       high  11283.0    349911.0    0.031238  \n",
       "I84           z4       high  12102.0    349092.0    0.033506  "
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quality_heritability_phenos[display_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "included-montreal",
   "metadata": {},
   "outputs": [],
   "source": [
    "quality_heritability_phenos.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "determined-stomach",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "description         Diagnoses - main ICD10: I84 Haemorrhoids\n",
       "h2_liability                                        0.057475\n",
       "h2_liability_se                                     0.013482\n",
       "h2_observed                                         0.009864\n",
       "h2_observed_se                                      0.002314\n",
       "h2_z                                                4.263044\n",
       "h2_p                                                 0.00001\n",
       "h2_sig                                                    z4\n",
       "confidence                                              high\n",
       "notes                                                    NaN\n",
       "intercept                                           1.018834\n",
       "intercept_se                                        0.008843\n",
       "intercept_z                                         2.129842\n",
       "intercept_p                                         0.016592\n",
       "lambdaGC                                            1.074945\n",
       "mean_chi2                                           1.085477\n",
       "ratio                                               0.220335\n",
       "ratio_se                                            0.103451\n",
       "n                                                     361194\n",
       "Neff                                                46786.06\n",
       "variable_type                                         binary\n",
       "isBinary                                                True\n",
       "n_cases                                              12102.0\n",
       "n_controls                                          349092.0\n",
       "prevalence                                          0.033506\n",
       "source                                                 icd10\n",
       "sex                                               both_sexes\n",
       "isNotPrimary                                           False\n",
       "isBadPower                                             False\n",
       "isLowNeff                                              False\n",
       "isMidNeff                                              False\n",
       "isExtremeSE                                            False\n",
       "isHighSE                                               False\n",
       "isSexBias                                              False\n",
       "isBadOrdinal                                           False\n",
       "isNumericOrdinal                                       False\n",
       "Name: I84, dtype: object"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno = quality_heritability_phenos.loc[\"I84\"]\n",
    "pheno"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "major-bosnia",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "meaningful-blowing",
   "metadata": {},
   "source": [
    "### Output phenotype matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "minute-tract",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/ahoang/Documents/MIT_classes/DeepLearning_Spring21/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/io/data_catalog.py:402: DtypeWarning: Columns (28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  result = func()\n"
     ]
    }
   ],
   "source": [
    "#export\n",
    "logging.warning(\"Loading ICD phenotype matrix, this might take a while\")\n",
    "icd10_pheno_matrix = catalog_all.load(\"ICD10_pheno_matrix\")\n",
    "\n",
    "\n",
    "#get the first 3 character of ICD code\n",
    "icd10_primary_cols = icd10_pheno_matrix.columns[icd10_pheno_matrix.columns.str.contains(\"primary\")]\n",
    "icd10_pheno_matrix = icd10_pheno_matrix.astype(str).apply(lambda x: x.str.slice(0,3))\n",
    "\n",
    "logging.warning(\"Finished loading ICD10 matrix\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "rural-librarian",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "def get_phenotype(icd10_codes: Union[str, List[str]] =\"I84\", samples:pd.Series=None):\n",
    "    \"\"\"\n",
    "    if samples argument is provided from genetic file, then find common set of samples and output ordered phenotype\n",
    "    \"\"\"\n",
    "    icd10_codes = [icd10_codes] if not isinstance(icd10_codes, list) else icd10_codes\n",
    "    pheno_df_list = [icd10_pheno_matrix[icd10_primary_cols].isin([icd10_code]).any(axis=1).astype(int) for icd10_code in icd10_codes]\n",
    "    pheno_df = pd.concat(pheno_df_list, axis=1)\n",
    "    pheno_df.columns = icd10_codes\n",
    "    \n",
    "    if samples is not None:\n",
    "        geno_pheno_sample_index_mask = np.isin(samples.values.astype(int), pheno_df.index)\n",
    "        pheno_geno_samples_common_set = samples.values[geno_pheno_sample_index_mask].astype(int)\n",
    "        pheno_df_ordered = pheno_df.loc[list(pheno_geno_samples_common_set), :]\n",
    "        pheno_df_ordered = pheno_df_ordered.loc[~pheno_df_ordered.index.duplicated(keep=\"first\"),:]\n",
    "        sample_index = np.argwhere(geno_pheno_sample_index_mask).reshape(-1)\n",
    "        return sample_index, pheno_df_ordered\n",
    "    return None, pheno_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "russian-virginia",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>I84</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f.eid</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1000025</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000038</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000042</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000056</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000061</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873158</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873167</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873175</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873180</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873199</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>357682 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         I84\n",
       "f.eid       \n",
       "1000025    0\n",
       "1000038    0\n",
       "1000042    0\n",
       "1000056    0\n",
       "1000061    0\n",
       "...      ...\n",
       "5873158    0\n",
       "5873167    0\n",
       "5873175    0\n",
       "5873180    0\n",
       "5873199    0\n",
       "\n",
       "[357682 rows x 1 columns]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_phenotype(icd10_codes = \"I84\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "intensive-evaluation",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>I84</th>\n",
       "      <th>R07</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>f.eid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1000025</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000038</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000042</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000056</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1000061</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873158</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873167</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873175</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873180</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5873199</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>357682 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         I84  R07\n",
       "f.eid            \n",
       "1000025    0    0\n",
       "1000038    0    0\n",
       "1000042    0    0\n",
       "1000056    0    0\n",
       "1000061    0    0\n",
       "...      ...  ...\n",
       "5873158    0    0\n",
       "5873167    0    0\n",
       "5873175    0    0\n",
       "5873180    0    0\n",
       "5873199    0    0\n",
       "\n",
       "[357682 rows x 2 columns]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_phenotype(icd10_codes = [\"I84\", \"R07\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "absent-ancient",
   "metadata": {},
   "source": [
    "---\n",
    "## Get SNPs used in GWAS for trait"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "passive-silver",
   "metadata": {},
   "outputs": [],
   "source": [
    "#export\n",
    "\n",
    "def get_GWAS_snps_for_trait(phenotype_code= \"I84\", chromosome:Union[int, List[int]] = 21, sort_val_cols_list: List[str] = [\"pval\"], ascending_bool_list: List[bool] = [True], id_only= True):\n",
    "    chromosome = [chromosome] if not isinstance(chromosome, list) else chromosome\n",
    "    chromosome_str = [f\"{single_chromosome}:\" for single_chromosome in chromosome]\n",
    "    gwas_result_df = read_csv_compressed(get_GWAS_result_link(phenotype_code=phenotype_code))\n",
    "    gwas_result_df = gwas_result_df.loc[gwas_result_df[\"variant\"].str.startswith(tuple(chromosome_str)), :]\n",
    "    gwas_result_df = gwas_result_df.reset_index(drop=True).reset_index().rename(columns = {\"index\":\"position_rank\"})\n",
    "    gwas_result_df = gwas_result_df.sort_values(sort_val_cols_list, ascending = ascending_bool_list)\n",
    "    variant_id_df = gwas_result_df[\"variant\"].str.split(\":\",expand=True)\n",
    "    variant_id_df[\"chr1_4\"] =variant_id_df[[1,2,3]].apply(\"_\".join, axis=1)\n",
    "    variant_id_df[1] = variant_id_df[1].astype(int)\n",
    "    gwas_result_df[[\"chr\", \"position\", \"major_allele\"]] = variant_id_df[[0, 1, 2]]\n",
    "    gwas_result_df[\"full_id\"] =  variant_id_df[[0, \"chr1_4\"]].apply(\":\".join, axis=1)\n",
    "    \n",
    "    if id_only:\n",
    "        return gwas_result_df[\"full_id\"].values\n",
    "    else:\n",
    "        return gwas_result_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "divine-bearing",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['21:46875774_C_T', '21:30968890_C_T', '21:43767602_A_G', ...,\n",
       "       '21:46876064_G_C', '21:47773893_C_G', '21:47831648_C_T'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_GWAS_snps_for_trait()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "alone-authentication",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position_rank</th>\n",
       "      <th>variant</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>minor_AF</th>\n",
       "      <th>expected_case_minor_AC</th>\n",
       "      <th>low_confidence_variant</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>AC</th>\n",
       "      <th>ytx</th>\n",
       "      <th>beta</th>\n",
       "      <th>se</th>\n",
       "      <th>tstat</th>\n",
       "      <th>pval</th>\n",
       "      <th>chr</th>\n",
       "      <th>position</th>\n",
       "      <th>major_allele</th>\n",
       "      <th>full_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46739</th>\n",
       "      <td>46739</td>\n",
       "      <td>21:23683958:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>0.448599</td>\n",
       "      <td>11373.8000</td>\n",
       "      <td>False</td>\n",
       "      <td>361194</td>\n",
       "      <td>324062.000</td>\n",
       "      <td>11381.3000</td>\n",
       "      <td>1.745250e-09</td>\n",
       "      <td>0.000435</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.999997</td>\n",
       "      <td>21</td>\n",
       "      <td>23683958</td>\n",
       "      <td>G</td>\n",
       "      <td>21:23683958_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156679</th>\n",
       "      <td>156679</td>\n",
       "      <td>21:44501642:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.012528</td>\n",
       "      <td>317.6430</td>\n",
       "      <td>False</td>\n",
       "      <td>361194</td>\n",
       "      <td>9050.310</td>\n",
       "      <td>315.2900</td>\n",
       "      <td>-1.938640e-08</td>\n",
       "      <td>0.002030</td>\n",
       "      <td>-0.000010</td>\n",
       "      <td>0.999992</td>\n",
       "      <td>21</td>\n",
       "      <td>44501642</td>\n",
       "      <td>C</td>\n",
       "      <td>21:44501642_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57576</th>\n",
       "      <td>57576</td>\n",
       "      <td>21:25570916:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.006228</td>\n",
       "      <td>157.8990</td>\n",
       "      <td>False</td>\n",
       "      <td>361194</td>\n",
       "      <td>4498.860</td>\n",
       "      <td>158.5530</td>\n",
       "      <td>-5.141850e-08</td>\n",
       "      <td>0.002875</td>\n",
       "      <td>-0.000018</td>\n",
       "      <td>0.999986</td>\n",
       "      <td>21</td>\n",
       "      <td>25570916</td>\n",
       "      <td>T</td>\n",
       "      <td>21:25570916_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17572</th>\n",
       "      <td>17572</td>\n",
       "      <td>21:18598893:A:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001108</td>\n",
       "      <td>28.0995</td>\n",
       "      <td>False</td>\n",
       "      <td>361194</td>\n",
       "      <td>800.612</td>\n",
       "      <td>28.0941</td>\n",
       "      <td>1.672180e-07</td>\n",
       "      <td>0.006937</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.999981</td>\n",
       "      <td>21</td>\n",
       "      <td>18598893</td>\n",
       "      <td>A</td>\n",
       "      <td>21:18598893_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89085</th>\n",
       "      <td>89085</td>\n",
       "      <td>21:31934274:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.011455</td>\n",
       "      <td>290.4320</td>\n",
       "      <td>False</td>\n",
       "      <td>361194</td>\n",
       "      <td>8275.000</td>\n",
       "      <td>288.0000</td>\n",
       "      <td>7.702210e-08</td>\n",
       "      <td>0.002036</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.999970</td>\n",
       "      <td>21</td>\n",
       "      <td>31934274</td>\n",
       "      <td>T</td>\n",
       "      <td>21:31934274_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161896</th>\n",
       "      <td>161896</td>\n",
       "      <td>21:45545974:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>45545974</td>\n",
       "      <td>C</td>\n",
       "      <td>21:45545974_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165184</th>\n",
       "      <td>165184</td>\n",
       "      <td>21:46067181:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>46067181</td>\n",
       "      <td>G</td>\n",
       "      <td>21:46067181_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170601</th>\n",
       "      <td>170601</td>\n",
       "      <td>21:46876064:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>46876064</td>\n",
       "      <td>G</td>\n",
       "      <td>21:46876064_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176072</th>\n",
       "      <td>176072</td>\n",
       "      <td>21:47773893:C:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>47773893</td>\n",
       "      <td>C</td>\n",
       "      <td>21:47773893_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176489</th>\n",
       "      <td>176489</td>\n",
       "      <td>21:47831648:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>47831648</td>\n",
       "      <td>C</td>\n",
       "      <td>21:47831648_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>178155 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        position_rank          variant minor_allele  minor_AF  \\\n",
       "46739           46739  21:23683958:G:A            A  0.448599   \n",
       "156679         156679  21:44501642:C:T            T  0.012528   \n",
       "57576           57576  21:25570916:T:C            C  0.006228   \n",
       "17572           17572  21:18598893:A:C            C  0.001108   \n",
       "89085           89085  21:31934274:T:C            C  0.011455   \n",
       "...               ...              ...          ...       ...   \n",
       "161896         161896  21:45545974:C:T            T  0.000000   \n",
       "165184         165184  21:46067181:G:C            C  0.000000   \n",
       "170601         170601  21:46876064:G:C            C  0.000000   \n",
       "176072         176072  21:47773893:C:G            G  0.000000   \n",
       "176489         176489  21:47831648:C:T            T  0.000000   \n",
       "\n",
       "        expected_case_minor_AC  low_confidence_variant  n_complete_samples  \\\n",
       "46739               11373.8000                   False              361194   \n",
       "156679                317.6430                   False              361194   \n",
       "57576                 157.8990                   False              361194   \n",
       "17572                  28.0995                   False              361194   \n",
       "89085                 290.4320                   False              361194   \n",
       "...                        ...                     ...                 ...   \n",
       "161896                  0.0000                    True              361194   \n",
       "165184                  0.0000                    True              361194   \n",
       "170601                  0.0000                    True              361194   \n",
       "176072                  0.0000                    True              361194   \n",
       "176489                  0.0000                    True              361194   \n",
       "\n",
       "                AC         ytx          beta        se     tstat      pval  \\\n",
       "46739   324062.000  11381.3000  1.745250e-09  0.000435  0.000004  0.999997   \n",
       "156679    9050.310    315.2900 -1.938640e-08  0.002030 -0.000010  0.999992   \n",
       "57576     4498.860    158.5530 -5.141850e-08  0.002875 -0.000018  0.999986   \n",
       "17572      800.612     28.0941  1.672180e-07  0.006937  0.000024  0.999981   \n",
       "89085     8275.000    288.0000  7.702210e-08  0.002036  0.000038  0.999970   \n",
       "...            ...         ...           ...       ...       ...       ...   \n",
       "161896       0.000      0.0000           NaN       NaN       NaN       NaN   \n",
       "165184       0.000      0.0000           NaN       NaN       NaN       NaN   \n",
       "170601       0.000      0.0000           NaN       NaN       NaN       NaN   \n",
       "176072       0.000      0.0000           NaN       NaN       NaN       NaN   \n",
       "176489       0.000      0.0000           NaN       NaN       NaN       NaN   \n",
       "\n",
       "       chr  position major_allele          full_id  \n",
       "46739   21  23683958            G  21:23683958_G_A  \n",
       "156679  21  44501642            C  21:44501642_C_T  \n",
       "57576   21  25570916            T  21:25570916_T_C  \n",
       "17572   21  18598893            A  21:18598893_A_C  \n",
       "89085   21  31934274            T  21:31934274_T_C  \n",
       "...     ..       ...          ...              ...  \n",
       "161896  21  45545974            C  21:45545974_C_T  \n",
       "165184  21  46067181            G  21:46067181_G_C  \n",
       "170601  21  46876064            G  21:46876064_G_C  \n",
       "176072  21  47773893            C  21:47773893_C_G  \n",
       "176489  21  47831648            C  21:47831648_C_T  \n",
       "\n",
       "[178155 rows x 17 columns]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_GWAS_snps_for_trait(id_only=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "471e44d1-339a-467f-bd4f-c85fb5db346a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position_rank</th>\n",
       "      <th>variant</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>minor_AF</th>\n",
       "      <th>expected_case_minor_AC</th>\n",
       "      <th>low_confidence_variant</th>\n",
       "      <th>n_complete_samples</th>\n",
       "      <th>AC</th>\n",
       "      <th>ytx</th>\n",
       "      <th>beta</th>\n",
       "      <th>se</th>\n",
       "      <th>tstat</th>\n",
       "      <th>pval</th>\n",
       "      <th>chr</th>\n",
       "      <th>position</th>\n",
       "      <th>major_allele</th>\n",
       "      <th>full_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>466355</th>\n",
       "      <td>466355</td>\n",
       "      <td>21:46875774:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>6.791200e-06</td>\n",
       "      <td>0.172184</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>4.905880</td>\n",
       "      <td>2.117650</td>\n",
       "      <td>0.667492</td>\n",
       "      <td>0.107762</td>\n",
       "      <td>6.19411</td>\n",
       "      <td>5.867920e-10</td>\n",
       "      <td>21</td>\n",
       "      <td>46875774</td>\n",
       "      <td>C</td>\n",
       "      <td>21:46875774_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194304</th>\n",
       "      <td>194304</td>\n",
       "      <td>20:43850800:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>1.737160e-07</td>\n",
       "      <td>0.004404</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.125490</td>\n",
       "      <td>0.086275</td>\n",
       "      <td>12.826500</td>\n",
       "      <td>2.294580</td>\n",
       "      <td>5.58991</td>\n",
       "      <td>2.273570e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>43850800</td>\n",
       "      <td>C</td>\n",
       "      <td>20:43850800_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149090</th>\n",
       "      <td>149090</td>\n",
       "      <td>20:33444611:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>1.644870e-06</td>\n",
       "      <td>0.041704</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>1.188240</td>\n",
       "      <td>0.568627</td>\n",
       "      <td>2.003390</td>\n",
       "      <td>0.359746</td>\n",
       "      <td>5.56892</td>\n",
       "      <td>2.565050e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>33444611</td>\n",
       "      <td>C</td>\n",
       "      <td>20:33444611_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83527</th>\n",
       "      <td>83527</td>\n",
       "      <td>20:16253954:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>8.088640e-07</td>\n",
       "      <td>0.020508</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.584314</td>\n",
       "      <td>0.533333</td>\n",
       "      <td>1.910680</td>\n",
       "      <td>0.354733</td>\n",
       "      <td>5.38626</td>\n",
       "      <td>7.198550e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>16253954</td>\n",
       "      <td>A</td>\n",
       "      <td>20:16253954_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183263</th>\n",
       "      <td>183263</td>\n",
       "      <td>20:41514559:A:G</td>\n",
       "      <td>G</td>\n",
       "      <td>3.257170e-08</td>\n",
       "      <td>0.000826</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.023529</td>\n",
       "      <td>0.011765</td>\n",
       "      <td>88.836800</td>\n",
       "      <td>16.586000</td>\n",
       "      <td>5.35613</td>\n",
       "      <td>8.507440e-08</td>\n",
       "      <td>20</td>\n",
       "      <td>41514559</td>\n",
       "      <td>A</td>\n",
       "      <td>20:41514559_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457657</th>\n",
       "      <td>457657</td>\n",
       "      <td>21:45545974:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>45545974</td>\n",
       "      <td>C</td>\n",
       "      <td>21:45545974_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460945</th>\n",
       "      <td>460945</td>\n",
       "      <td>21:46067181:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>46067181</td>\n",
       "      <td>G</td>\n",
       "      <td>21:46067181_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>466362</th>\n",
       "      <td>466362</td>\n",
       "      <td>21:46876064:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>46876064</td>\n",
       "      <td>G</td>\n",
       "      <td>21:46876064_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471833</th>\n",
       "      <td>471833</td>\n",
       "      <td>21:47773893:C:G</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>47773893</td>\n",
       "      <td>C</td>\n",
       "      <td>21:47773893_C_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472250</th>\n",
       "      <td>472250</td>\n",
       "      <td>21:47831648:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>True</td>\n",
       "      <td>361194</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21</td>\n",
       "      <td>47831648</td>\n",
       "      <td>C</td>\n",
       "      <td>21:47831648_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>473916 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        position_rank          variant minor_allele      minor_AF  \\\n",
       "466355         466355  21:46875774:C:T            T  6.791200e-06   \n",
       "194304         194304  20:43850800:C:T            T  1.737160e-07   \n",
       "149090         149090  20:33444611:C:T            T  1.644870e-06   \n",
       "83527           83527  20:16253954:A:G            G  8.088640e-07   \n",
       "183263         183263  20:41514559:A:G            G  3.257170e-08   \n",
       "...               ...              ...          ...           ...   \n",
       "457657         457657  21:45545974:C:T            T  0.000000e+00   \n",
       "460945         460945  21:46067181:G:C            C  0.000000e+00   \n",
       "466362         466362  21:46876064:G:C            C  0.000000e+00   \n",
       "471833         471833  21:47773893:C:G            G  0.000000e+00   \n",
       "472250         472250  21:47831648:C:T            T  0.000000e+00   \n",
       "\n",
       "        expected_case_minor_AC  low_confidence_variant  n_complete_samples  \\\n",
       "466355                0.172184                    True              361194   \n",
       "194304                0.004404                    True              361194   \n",
       "149090                0.041704                    True              361194   \n",
       "83527                 0.020508                    True              361194   \n",
       "183263                0.000826                    True              361194   \n",
       "...                        ...                     ...                 ...   \n",
       "457657                0.000000                    True              361194   \n",
       "460945                0.000000                    True              361194   \n",
       "466362                0.000000                    True              361194   \n",
       "471833                0.000000                    True              361194   \n",
       "472250                0.000000                    True              361194   \n",
       "\n",
       "              AC       ytx       beta         se    tstat          pval chr  \\\n",
       "466355  4.905880  2.117650   0.667492   0.107762  6.19411  5.867920e-10  21   \n",
       "194304  0.125490  0.086275  12.826500   2.294580  5.58991  2.273570e-08  20   \n",
       "149090  1.188240  0.568627   2.003390   0.359746  5.56892  2.565050e-08  20   \n",
       "83527   0.584314  0.533333   1.910680   0.354733  5.38626  7.198550e-08  20   \n",
       "183263  0.023529  0.011765  88.836800  16.586000  5.35613  8.507440e-08  20   \n",
       "...          ...       ...        ...        ...      ...           ...  ..   \n",
       "457657  0.000000  0.000000        NaN        NaN      NaN           NaN  21   \n",
       "460945  0.000000  0.000000        NaN        NaN      NaN           NaN  21   \n",
       "466362  0.000000  0.000000        NaN        NaN      NaN           NaN  21   \n",
       "471833  0.000000  0.000000        NaN        NaN      NaN           NaN  21   \n",
       "472250  0.000000  0.000000        NaN        NaN      NaN           NaN  21   \n",
       "\n",
       "        position major_allele          full_id  \n",
       "466355  46875774            C  21:46875774_C_T  \n",
       "194304  43850800            C  20:43850800_C_T  \n",
       "149090  33444611            C  20:33444611_C_T  \n",
       "83527   16253954            A  20:16253954_A_G  \n",
       "183263  41514559            A  20:41514559_A_G  \n",
       "...          ...          ...              ...  \n",
       "457657  45545974            C  21:45545974_C_T  \n",
       "460945  46067181            G  21:46067181_G_C  \n",
       "466362  46876064            G  21:46876064_G_C  \n",
       "471833  47773893            C  21:47773893_C_G  \n",
       "472250  47831648            C  21:47831648_C_T  \n",
       "\n",
       "[473916 rows x 17 columns]"
      ]
     },
     "execution_count": null,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = get_GWAS_snps_for_trait(chromosome = [20, 21], id_only=False)\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a965bb29-1847-4972-b66a-5d4c59078f2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "assert sorted(test[\"chr\"].unique()) == ['20', '21']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
