---

title: Functions to extract phenotype info


keywords: fastai
sidebar: home_sidebar

summary: "API details."
description: "API details."
nb_path: "notebooks/package/phenotypes.ipynb"
---
<!--

#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: notebooks/package/phenotypes.ipynb
# command to build the docs after a change: nbdev_build_docs

-->

<div class="container" id="notebook-container">
        
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/extras/datasets/pandas/csv_dataset.py:158: ParserWarning: Falling back to the &#39;python&#39; engine because the &#39;c&#39; engine does not support regex separators (separators &gt; 1 char and different from &#39;\s+&#39; are interpreted as regex); you can avoid this warning by specifying engine=&#39;python&#39;.
  return pd.read_csv(fs_file, **self._load_args)
WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/combinatorial_gwas/phenotypes/__init__.py:24: UserWarning: registration of accessor &lt;class &#39;combinatorial_gwas.phenotypes.__init__.QueryDataframe&#39;&gt; under name &#39;pheno&#39; for type &lt;class &#39;pandas.core.frame.DataFrame&#39;&gt; is overriding a preexisting attribute with the same name.
  class QueryDataframe():
WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h2 id="QueryDataframe" class="doc_header"><code>class</code> <code>QueryDataframe</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/phenotypes/__init__.py#L24" class="source_link" style="float:right">[source]</a></h2><blockquote><p><code>QueryDataframe</code>(<strong><code>df</code></strong>:<code>DataFrame</code>)</p>
</blockquote>
<p>QueryDataframe(df: pandas.core.frame.DataFrame)</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>{&#39;example_test_data_ratio&#39;: 0.2,
 &#39;example_num_train_iter&#39;: 10000,
 &#39;example_learning_rate&#39;: 0.01,
 &#39;template_gwas_result_file_link&#39;: &#39;https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/{phenotype_code}.gwas.imputed_v3.both_sexes.tsv.bgz&#39;,
 &#39;genetic_file_path_template&#39;: &#39;/lab/corradin_biobank/Raw_UKB_downloads/BGEN/ukb_imp_chr{chrom_number}_v3.bgen&#39;,
 &#39;sample_file_template&#39;: &#39;/lab/corradin_biobank/Raw_UKB_downloads/sample_files/ukb45624_imp_chr21_v3_s487275.sample&#39;,
 &#39;data_dict_file&#39;: &#39;/lab/corradin_biobank/Phenotypes//Data_Dictionary_Showcase.csv&#39;,
 &#39;pheno_file&#39;: &#39;/lab/corradin_biobank/samples//neale_gwas_both_sexes_parquet/&#39;,
 &#39;coding_file_path_template&#39;: &#39;/lab/corradin_biobank/Phenotypes//data_codes/datacode-*.tsv&#39;,
 &#39;id_col&#39;: &#39;f.eid&#39;}</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>[&#39;example_iris_data&#39;,
 &#39;globals_test&#39;,
 &#39;disease_summary&#39;,
 &#39;sample_ids&#39;,
 &#39;chr_22&#39;,
 &#39;ICD10_pheno_matrix&#39;,
 &#39;phenotypes_info_Neale&#39;,
 &#39;GWAS_results_links_df&#39;,
 &#39;heritability_trait_level_summary&#39;,
 &#39;heritability_trait_level_partitioned_z4_sig&#39;]</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">sample_ids</span> <span class="o">=</span> <span class="n">catalog_all</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="s2">&quot;sample_ids&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Find-high-heritability-traits">Find high heritability traits<a class="anchor-link" href="#Find-high-heritability-traits"> </a></h3>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/extras/datasets/pandas/csv_dataset.py:158: ParserWarning: Falling back to the &#39;python&#39; engine because the &#39;c&#39; engine does not support regex separators (separators &gt; 1 char and different from &#39;\s+&#39; are interpreted as regex); you can avoid this warning by specifying engine=&#39;python&#39;.
  return pd.read_csv(fs_file, **self._load_args)
</pre>
</div>
</div>

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>phenotype</th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_liability_se</th>
      <th>h2_observed</th>
      <th>h2_observed_se</th>
      <th>h2_z</th>
      <th>h2_p</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>...</th>
      <th>sex</th>
      <th>isNotPrimary</th>
      <th>isBadPower</th>
      <th>isLowNeff</th>
      <th>isMidNeff</th>
      <th>isExtremeSE</th>
      <th>isHighSE</th>
      <th>isSexBias</th>
      <th>isBadOrdinal</th>
      <th>isNumericOrdinal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100001_irnt</td>
      <td>Food weight</td>
      <td>0.068818</td>
      <td>0.016857</td>
      <td>0.068818</td>
      <td>0.016857</td>
      <td>4.082528</td>
      <td>0.000022</td>
      <td>z4</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100002_irnt</td>
      <td>Energy</td>
      <td>0.064783</td>
      <td>0.015784</td>
      <td>0.064783</td>
      <td>0.015784</td>
      <td>4.104249</td>
      <td>0.000020</td>
      <td>z4</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100003_irnt</td>
      <td>Protein</td>
      <td>0.034543</td>
      <td>0.014619</td>
      <td>0.034543</td>
      <td>0.014619</td>
      <td>2.362892</td>
      <td>0.009066</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>3</th>
      <td>100004_irnt</td>
      <td>Fat</td>
      <td>0.055197</td>
      <td>0.018265</td>
      <td>0.055197</td>
      <td>0.018265</td>
      <td>3.021937</td>
      <td>0.001256</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4</th>
      <td>100005_irnt</td>
      <td>Carbohydrate</td>
      <td>0.051744</td>
      <td>0.015357</td>
      <td>0.051744</td>
      <td>0.015357</td>
      <td>3.369359</td>
      <td>0.000377</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
  </tbody>
</table>
<p>5 rows Ã— 37 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">heritability_Neale</span><span class="o">.</span><span class="n">columns</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>Index([&#39;phenotype&#39;, &#39;description&#39;, &#39;h2_liability&#39;, &#39;h2_liability_se&#39;,
       &#39;h2_observed&#39;, &#39;h2_observed_se&#39;, &#39;h2_z&#39;, &#39;h2_p&#39;, &#39;h2_sig&#39;, &#39;confidence&#39;,
       &#39;notes&#39;, &#39;intercept&#39;, &#39;intercept_se&#39;, &#39;intercept_z&#39;, &#39;intercept_p&#39;,
       &#39;lambdaGC&#39;, &#39;mean_chi2&#39;, &#39;ratio&#39;, &#39;ratio_se&#39;, &#39;n&#39;, &#39;Neff&#39;,
       &#39;variable_type&#39;, &#39;isBinary&#39;, &#39;n_cases&#39;, &#39;n_controls&#39;, &#39;prevalence&#39;,
       &#39;source&#39;, &#39;sex&#39;, &#39;isNotPrimary&#39;, &#39;isBadPower&#39;, &#39;isLowNeff&#39;, &#39;isMidNeff&#39;,
       &#39;isExtremeSE&#39;, &#39;isHighSE&#39;, &#39;isSexBias&#39;, &#39;isBadOrdinal&#39;,
       &#39;isNumericOrdinal&#39;],
      dtype=&#39;object&#39;)</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">heritability_Neale</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s2">&quot;h2_sig == [&#39;z7&#39;, &#39;z4&#39;]&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;h2_liability&quot;</span><span class="p">,</span> <span class="n">ascending</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">60</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>phenotype</th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_liability_se</th>
      <th>h2_observed</th>
      <th>h2_observed_se</th>
      <th>h2_z</th>
      <th>h2_p</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>...</th>
      <th>sex</th>
      <th>isNotPrimary</th>
      <th>isBadPower</th>
      <th>isLowNeff</th>
      <th>isMidNeff</th>
      <th>isExtremeSE</th>
      <th>isHighSE</th>
      <th>isSexBias</th>
      <th>isBadOrdinal</th>
      <th>isNumericOrdinal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>2543</th>
      <td>50_irnt</td>
      <td>Standing height</td>
      <td>0.485223</td>
      <td>0.021812</td>
      <td>0.485223</td>
      <td>0.021812</td>
      <td>22.246180</td>
      <td>6.140928e-110</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2554</th>
      <td>5097_irnt</td>
      <td>6mm weak meridian (left)</td>
      <td>0.433908</td>
      <td>0.038336</td>
      <td>0.433908</td>
      <td>0.038336</td>
      <td>11.318693</td>
      <td>5.302052e-30</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1293</th>
      <td>20153_irnt</td>
      <td>Forced expiratory volume in 1-second (FEV1), p...</td>
      <td>0.433270</td>
      <td>0.023709</td>
      <td>0.433270</td>
      <td>0.023709</td>
      <td>18.274839</td>
      <td>6.563795e-75</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2579</th>
      <td>5134_irnt</td>
      <td>6mm strong meridian (left)</td>
      <td>0.427685</td>
      <td>0.038038</td>
      <td>0.427685</td>
      <td>0.038038</td>
      <td>11.243581</td>
      <td>1.245346e-29</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4155</th>
      <td>2395_4</td>
      <td>Hair/balding pattern: Pattern 4</td>
      <td>0.423410</td>
      <td>0.042975</td>
      <td>0.198793</td>
      <td>0.020177</td>
      <td>9.852372</td>
      <td>3.347251e-23</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>male</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2578</th>
      <td>5133_irnt</td>
      <td>6mm strong meridian (right)</td>
      <td>0.423025</td>
      <td>0.036995</td>
      <td>0.423025</td>
      <td>0.036995</td>
      <td>11.434813</td>
      <td>1.400691e-30</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2555</th>
      <td>5098_irnt</td>
      <td>6mm weak meridian (right)</td>
      <td>0.418781</td>
      <td>0.037019</td>
      <td>0.418781</td>
      <td>0.037019</td>
      <td>11.312652</td>
      <td>5.680164e-30</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2302</th>
      <td>30100_irnt</td>
      <td>Mean platelet (thrombocyte) volume</td>
      <td>0.406220</td>
      <td>0.057866</td>
      <td>0.406220</td>
      <td>0.057866</td>
      <td>7.019956</td>
      <td>1.109691e-12</td>
      <td>z7</td>
      <td>medium</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2553</th>
      <td>5096_irnt</td>
      <td>3mm weak meridian (left)</td>
      <td>0.393818</td>
      <td>0.038762</td>
      <td>0.393818</td>
      <td>0.038762</td>
      <td>10.160001</td>
      <td>1.495374e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4152</th>
      <td>2395_1</td>
      <td>Hair/balding pattern: Pattern 1</td>
      <td>0.387338</td>
      <td>0.036491</td>
      <td>0.227732</td>
      <td>0.021454</td>
      <td>10.614754</td>
      <td>1.272325e-26</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>male</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2556</th>
      <td>5099_irnt</td>
      <td>3mm weak meridian (right)</td>
      <td>0.387322</td>
      <td>0.040353</td>
      <td>0.387322</td>
      <td>0.040353</td>
      <td>9.598457</td>
      <td>4.057488e-22</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2580</th>
      <td>5135_irnt</td>
      <td>3mm strong meridian (left)</td>
      <td>0.385232</td>
      <td>0.037401</td>
      <td>0.385232</td>
      <td>0.037401</td>
      <td>10.300087</td>
      <td>3.519882e-25</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2577</th>
      <td>5132_irnt</td>
      <td>3mm strong meridian (right)</td>
      <td>0.383519</td>
      <td>0.039018</td>
      <td>0.383519</td>
      <td>0.039018</td>
      <td>9.829196</td>
      <td>4.214439e-23</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1084</th>
      <td>20015_irnt</td>
      <td>Sitting height</td>
      <td>0.348691</td>
      <td>0.016679</td>
      <td>0.348691</td>
      <td>0.016679</td>
      <td>20.905813</td>
      <td>2.370294e-97</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2600</th>
      <td>5257_irnt</td>
      <td>Corneal resistance factor (right)</td>
      <td>0.335975</td>
      <td>0.028842</td>
      <td>0.335975</td>
      <td>0.028842</td>
      <td>11.648723</td>
      <td>1.164611e-31</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2448</th>
      <td>4105_irnt</td>
      <td>Heel bone mineral density (BMD) (left)</td>
      <td>0.334856</td>
      <td>0.033472</td>
      <td>0.334856</td>
      <td>0.033472</td>
      <td>10.004035</td>
      <td>7.315592e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2459</th>
      <td>4124_irnt</td>
      <td>Heel bone mineral density (BMD) (right)</td>
      <td>0.332595</td>
      <td>0.034420</td>
      <td>0.332595</td>
      <td>0.034420</td>
      <td>9.662767</td>
      <td>2.169904e-22</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2449</th>
      <td>4106_irnt</td>
      <td>Heel bone mineral density (BMD) T-score, autom...</td>
      <td>0.332009</td>
      <td>0.033339</td>
      <td>0.332009</td>
      <td>0.033339</td>
      <td>9.958575</td>
      <td>1.156770e-23</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2447</th>
      <td>4104_irnt</td>
      <td>Heel quantitative ultrasound index (QUI), dire...</td>
      <td>0.332008</td>
      <td>0.033339</td>
      <td>0.332008</td>
      <td>0.033339</td>
      <td>9.958647</td>
      <td>1.155933e-23</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2456</th>
      <td>4123_irnt</td>
      <td>Heel quantitative ultrasound index (QUI), dire...</td>
      <td>0.331454</td>
      <td>0.034208</td>
      <td>0.331454</td>
      <td>0.034208</td>
      <td>9.689506</td>
      <td>1.670712e-22</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2476</th>
      <td>4125_irnt</td>
      <td>Heel bone mineral density (BMD) T-score, autom...</td>
      <td>0.331446</td>
      <td>0.034206</td>
      <td>0.331446</td>
      <td>0.034206</td>
      <td>9.689696</td>
      <td>1.667599e-22</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2548</th>
      <td>5085_irnt</td>
      <td>Spherical power (left)</td>
      <td>0.328175</td>
      <td>0.020486</td>
      <td>0.328175</td>
      <td>0.020486</td>
      <td>16.019682</td>
      <td>4.656245e-58</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2547</th>
      <td>5084_irnt</td>
      <td>Spherical power (right)</td>
      <td>0.326352</td>
      <td>0.020095</td>
      <td>0.326352</td>
      <td>0.020095</td>
      <td>16.240512</td>
      <td>1.303684e-59</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2445</th>
      <td>4100_irnt</td>
      <td>Ankle spacing width (left)</td>
      <td>0.317641</td>
      <td>0.017658</td>
      <td>0.317641</td>
      <td>0.017658</td>
      <td>17.988582</td>
      <td>1.197038e-72</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2368</th>
      <td>3148_irnt</td>
      <td>Heel bone mineral density (BMD)</td>
      <td>0.317506</td>
      <td>0.031145</td>
      <td>0.317506</td>
      <td>0.031145</td>
      <td>10.194588</td>
      <td>1.048155e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2850</th>
      <td>78_irnt</td>
      <td>Heel bone mineral density (BMD) T-score, autom...</td>
      <td>0.315207</td>
      <td>0.031078</td>
      <td>0.315207</td>
      <td>0.031078</td>
      <td>10.142315</td>
      <td>1.792511e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2367</th>
      <td>3147_irnt</td>
      <td>Heel quantitative ultrasound index (QUI), dire...</td>
      <td>0.315203</td>
      <td>0.031079</td>
      <td>0.315203</td>
      <td>0.031079</td>
      <td>10.142120</td>
      <td>1.796082e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2450</th>
      <td>4119_irnt</td>
      <td>Ankle spacing width (right)</td>
      <td>0.312397</td>
      <td>0.016995</td>
      <td>0.312397</td>
      <td>0.016995</td>
      <td>18.381263</td>
      <td>9.279797e-76</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2604</th>
      <td>5265_irnt</td>
      <td>Corneal resistance factor (left)</td>
      <td>0.311305</td>
      <td>0.026667</td>
      <td>0.311305</td>
      <td>0.026667</td>
      <td>11.673911</td>
      <td>8.663491e-32</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2300</th>
      <td>30080_irnt</td>
      <td>Platelet count</td>
      <td>0.307884</td>
      <td>0.032764</td>
      <td>0.307884</td>
      <td>0.032764</td>
      <td>9.397034</td>
      <td>2.806142e-21</td>
      <td>z7</td>
      <td>medium</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2201</th>
      <td>23101_irnt</td>
      <td>Whole body fat-free mass</td>
      <td>0.302503</td>
      <td>0.011553</td>
      <td>0.302503</td>
      <td>0.011553</td>
      <td>26.184329</td>
      <td>2.004229e-151</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2202</th>
      <td>23102_irnt</td>
      <td>Whole body water mass</td>
      <td>0.302087</td>
      <td>0.011557</td>
      <td>0.302087</td>
      <td>0.011557</td>
      <td>26.138755</td>
      <td>6.614652e-151</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2365</th>
      <td>3143_irnt</td>
      <td>Ankle spacing width</td>
      <td>0.301831</td>
      <td>0.015535</td>
      <td>0.301831</td>
      <td>0.015535</td>
      <td>19.428648</td>
      <td>2.209243e-84</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2446</th>
      <td>4101_irnt</td>
      <td>Heel broadband ultrasound attenuation (left)</td>
      <td>0.298408</td>
      <td>0.031728</td>
      <td>0.298408</td>
      <td>0.031728</td>
      <td>9.405259</td>
      <td>2.595103e-21</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2228</th>
      <td>23129_irnt</td>
      <td>Trunk fat-free mass</td>
      <td>0.298052</td>
      <td>0.011476</td>
      <td>0.298052</td>
      <td>0.011476</td>
      <td>25.972560</td>
      <td>5.057101e-149</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2204</th>
      <td>23105_irnt</td>
      <td>Basal metabolic rate</td>
      <td>0.296321</td>
      <td>0.011322</td>
      <td>0.296321</td>
      <td>0.011322</td>
      <td>26.172150</td>
      <td>2.758118e-151</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2229</th>
      <td>23130_irnt</td>
      <td>Trunk predicted mass</td>
      <td>0.296296</td>
      <td>0.011430</td>
      <td>0.296296</td>
      <td>0.011430</td>
      <td>25.923194</td>
      <td>1.824027e-148</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2451</th>
      <td>4120_irnt</td>
      <td>Heel broadband ultrasound attenuation (right)</td>
      <td>0.294529</td>
      <td>0.031884</td>
      <td>0.294529</td>
      <td>0.031884</td>
      <td>9.237596</td>
      <td>1.260491e-20</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2366</th>
      <td>3144_irnt</td>
      <td>Heel Broadband ultrasound attenuation, direct ...</td>
      <td>0.288307</td>
      <td>0.028292</td>
      <td>0.288307</td>
      <td>0.028292</td>
      <td>10.190395</td>
      <td>1.094363e-24</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2675</th>
      <td>6138_1</td>
      <td>Qualifications: College or University degree</td>
      <td>0.285602</td>
      <td>0.009297</td>
      <td>0.168548</td>
      <td>0.005487</td>
      <td>30.719943</td>
      <td>1.541503e-207</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2212</th>
      <td>23113_irnt</td>
      <td>Leg fat-free mass (right)</td>
      <td>0.282356</td>
      <td>0.010881</td>
      <td>0.282356</td>
      <td>0.010881</td>
      <td>25.950270</td>
      <td>9.028024e-149</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2213</th>
      <td>23114_irnt</td>
      <td>Leg predicted mass (right)</td>
      <td>0.282212</td>
      <td>0.010856</td>
      <td>0.282212</td>
      <td>0.010856</td>
      <td>25.997054</td>
      <td>2.673464e-149</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2216</th>
      <td>23117_irnt</td>
      <td>Leg fat-free mass (left)</td>
      <td>0.281976</td>
      <td>0.010893</td>
      <td>0.281976</td>
      <td>0.010893</td>
      <td>25.887005</td>
      <td>4.664133e-148</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2217</th>
      <td>23118_irnt</td>
      <td>Leg predicted mass (left)</td>
      <td>0.281242</td>
      <td>0.010851</td>
      <td>0.281242</td>
      <td>0.010851</td>
      <td>25.919000</td>
      <td>2.033838e-148</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2599</th>
      <td>5256_irnt</td>
      <td>Corneal hysteresis (right)</td>
      <td>0.275428</td>
      <td>0.025017</td>
      <td>0.275428</td>
      <td>0.025017</td>
      <td>11.009806</td>
      <td>1.713706e-28</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2221</th>
      <td>23122_irnt</td>
      <td>Arm predicted mass (right)</td>
      <td>0.274811</td>
      <td>0.010110</td>
      <td>0.274811</td>
      <td>0.010110</td>
      <td>27.181847</td>
      <td>5.323478e-163</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2220</th>
      <td>23121_irnt</td>
      <td>Arm fat-free mass (right)</td>
      <td>0.274690</td>
      <td>0.010120</td>
      <td>0.274690</td>
      <td>0.010120</td>
      <td>27.142264</td>
      <td>1.562285e-162</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2224</th>
      <td>23125_irnt</td>
      <td>Arm fat-free mass (left)</td>
      <td>0.269922</td>
      <td>0.010019</td>
      <td>0.269922</td>
      <td>0.010019</td>
      <td>26.942153</td>
      <td>3.524759e-160</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2225</th>
      <td>23126_irnt</td>
      <td>Arm predicted mass (left)</td>
      <td>0.269056</td>
      <td>0.010015</td>
      <td>0.269056</td>
      <td>0.010015</td>
      <td>26.864994</td>
      <td>2.817778e-159</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2205</th>
      <td>23106_irnt</td>
      <td>Impedance of whole body</td>
      <td>0.268544</td>
      <td>0.009617</td>
      <td>0.268544</td>
      <td>0.009617</td>
      <td>27.923605</td>
      <td>6.897033e-172</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2295</th>
      <td>30040_irnt</td>
      <td>Mean corpuscular volume</td>
      <td>0.266674</td>
      <td>0.035533</td>
      <td>0.266674</td>
      <td>0.035533</td>
      <td>7.504958</td>
      <td>3.072409e-14</td>
      <td>z7</td>
      <td>medium</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2198</th>
      <td>23098_irnt</td>
      <td>Weight</td>
      <td>0.265599</td>
      <td>0.009805</td>
      <td>0.265599</td>
      <td>0.009805</td>
      <td>27.087627</td>
      <td>6.887055e-162</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1483</th>
      <td>21002_irnt</td>
      <td>Weight</td>
      <td>0.265415</td>
      <td>0.009815</td>
      <td>0.265415</td>
      <td>0.009815</td>
      <td>27.041975</td>
      <td>2.373356e-161</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4037</th>
      <td>6153_2</td>
      <td>Medication for cholesterol, blood pressure, di...</td>
      <td>0.258441</td>
      <td>0.018049</td>
      <td>0.118466</td>
      <td>0.008273</td>
      <td>14.319151</td>
      <td>8.307080e-47</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>female</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2301</th>
      <td>30090_irnt</td>
      <td>Platelet crit</td>
      <td>0.255684</td>
      <td>0.034362</td>
      <td>0.255684</td>
      <td>0.034362</td>
      <td>7.440785</td>
      <td>5.004445e-14</td>
      <td>z7</td>
      <td>medium</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2347</th>
      <td>30770_irnt</td>
      <td>IGF-1 (nmol/L)</td>
      <td>0.253212</td>
      <td>0.018468</td>
      <td>0.253212</td>
      <td>0.018468</td>
      <td>13.710800</td>
      <td>4.374420e-43</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2297</th>
      <td>30050_irnt</td>
      <td>Mean corpuscular haemoglobin</td>
      <td>0.253001</td>
      <td>0.033201</td>
      <td>0.253001</td>
      <td>0.033201</td>
      <td>7.620246</td>
      <td>1.265960e-14</td>
      <td>z7</td>
      <td>medium</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>True</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2207</th>
      <td>23108_irnt</td>
      <td>Impedance of leg (left)</td>
      <td>0.252192</td>
      <td>0.009544</td>
      <td>0.252192</td>
      <td>0.009544</td>
      <td>26.422957</td>
      <td>3.732895e-154</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2206</th>
      <td>23107_irnt</td>
      <td>Impedance of leg (right)</td>
      <td>0.251010</td>
      <td>0.009489</td>
      <td>0.251010</td>
      <td>0.009489</td>
      <td>26.453122</td>
      <td>1.679547e-154</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2203</th>
      <td>23104_irnt</td>
      <td>Body mass index (BMI)</td>
      <td>0.248880</td>
      <td>0.008376</td>
      <td>0.248880</td>
      <td>0.008376</td>
      <td>29.714182</td>
      <td>2.518128e-194</td>
      <td>z7</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
  </tbody>
</table>
<p>60 rows Ã— 37 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>n_cases</th>
      <th>n_controls</th>
      <th>prevalence</th>
    </tr>
    <tr>
      <th>phenotype</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>I83</th>
      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>
      <td>0.189299</td>
      <td>z7</td>
      <td>medium</td>
      <td>8763.0</td>
      <td>352431.0</td>
      <td>0.024261</td>
    </tr>
    <tr>
      <th>I25</th>
      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>
      <td>0.163823</td>
      <td>z7</td>
      <td>medium</td>
      <td>12769.0</td>
      <td>348425.0</td>
      <td>0.035352</td>
    </tr>
    <tr>
      <th>G56</th>
      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>
      <td>0.157519</td>
      <td>z7</td>
      <td>medium</td>
      <td>8130.0</td>
      <td>353064.0</td>
      <td>0.022509</td>
    </tr>
    <tr>
      <th>N81</th>
      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>
      <td>0.152223</td>
      <td>z4</td>
      <td>medium</td>
      <td>7511.0</td>
      <td>186663.0</td>
      <td>0.038682</td>
    </tr>
    <tr>
      <th>M16</th>
      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>
      <td>0.151297</td>
      <td>z4</td>
      <td>medium</td>
      <td>9136.0</td>
      <td>352058.0</td>
      <td>0.025294</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span><span class="p">[</span><span class="n">display_cols</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>n_cases</th>
      <th>n_controls</th>
      <th>prevalence</th>
    </tr>
    <tr>
      <th>phenotype</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>I83</th>
      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>
      <td>0.189299</td>
      <td>z7</td>
      <td>medium</td>
      <td>8763.0</td>
      <td>352431.0</td>
      <td>0.024261</td>
    </tr>
    <tr>
      <th>I25</th>
      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>
      <td>0.163823</td>
      <td>z7</td>
      <td>medium</td>
      <td>12769.0</td>
      <td>348425.0</td>
      <td>0.035352</td>
    </tr>
    <tr>
      <th>G56</th>
      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>
      <td>0.157519</td>
      <td>z7</td>
      <td>medium</td>
      <td>8130.0</td>
      <td>353064.0</td>
      <td>0.022509</td>
    </tr>
    <tr>
      <th>N81</th>
      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>
      <td>0.152223</td>
      <td>z4</td>
      <td>medium</td>
      <td>7511.0</td>
      <td>186663.0</td>
      <td>0.038682</td>
    </tr>
    <tr>
      <th>M16</th>
      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>
      <td>0.151297</td>
      <td>z4</td>
      <td>medium</td>
      <td>9136.0</td>
      <td>352058.0</td>
      <td>0.025294</td>
    </tr>
    <tr>
      <th>M17</th>
      <td>Diagnoses - main ICD10: M17 Gonarthrosis [arth...</td>
      <td>0.144360</td>
      <td>z7</td>
      <td>high</td>
      <td>11497.0</td>
      <td>349697.0</td>
      <td>0.031831</td>
    </tr>
    <tr>
      <th>I48</th>
      <td>Diagnoses - main ICD10: I48 Atrial fibrillatio...</td>
      <td>0.144326</td>
      <td>z4</td>
      <td>medium</td>
      <td>6356.0</td>
      <td>354838.0</td>
      <td>0.017597</td>
    </tr>
    <tr>
      <th>C44</th>
      <td>Diagnoses - main ICD10: C44 Other malignant ne...</td>
      <td>0.141495</td>
      <td>z4</td>
      <td>medium</td>
      <td>9086.0</td>
      <td>352108.0</td>
      <td>0.025155</td>
    </tr>
    <tr>
      <th>K80</th>
      <td>Diagnoses - main ICD10: K80 Cholelithiasis</td>
      <td>0.137236</td>
      <td>z4</td>
      <td>high</td>
      <td>10520.0</td>
      <td>350674.0</td>
      <td>0.029126</td>
    </tr>
    <tr>
      <th>I21</th>
      <td>Diagnoses - main ICD10: I21 Acute myocardial i...</td>
      <td>0.136628</td>
      <td>z4</td>
      <td>medium</td>
      <td>5948.0</td>
      <td>355246.0</td>
      <td>0.016468</td>
    </tr>
    <tr>
      <th>I20</th>
      <td>Diagnoses - main ICD10: I20 Angina pectoris</td>
      <td>0.129187</td>
      <td>z4</td>
      <td>medium</td>
      <td>6246.0</td>
      <td>354948.0</td>
      <td>0.017293</td>
    </tr>
    <tr>
      <th>K57</th>
      <td>Diagnoses - main ICD10: K57 Diverticular disea...</td>
      <td>0.129184</td>
      <td>z7</td>
      <td>high</td>
      <td>12662.0</td>
      <td>348532.0</td>
      <td>0.035056</td>
    </tr>
    <tr>
      <th>K40</th>
      <td>Diagnoses - main ICD10: K40 Inguinal hernia</td>
      <td>0.120213</td>
      <td>z7</td>
      <td>medium</td>
      <td>13147.0</td>
      <td>348047.0</td>
      <td>0.036399</td>
    </tr>
    <tr>
      <th>D12</th>
      <td>Diagnoses - main ICD10: D12 Benign neoplasm of...</td>
      <td>0.118093</td>
      <td>z4</td>
      <td>medium</td>
      <td>8877.0</td>
      <td>352317.0</td>
      <td>0.024577</td>
    </tr>
    <tr>
      <th>M54</th>
      <td>Diagnoses - main ICD10: M54 Dorsalgia</td>
      <td>0.114167</td>
      <td>z7</td>
      <td>medium</td>
      <td>8361.0</td>
      <td>352833.0</td>
      <td>0.023148</td>
    </tr>
    <tr>
      <th>C50</th>
      <td>Diagnoses - main ICD10: C50 Malignant neoplasm...</td>
      <td>0.110280</td>
      <td>z4</td>
      <td>medium</td>
      <td>8304.0</td>
      <td>352890.0</td>
      <td>0.022990</td>
    </tr>
    <tr>
      <th>K22</th>
      <td>Diagnoses - main ICD10: K22 Other diseases of ...</td>
      <td>0.107578</td>
      <td>z4</td>
      <td>medium</td>
      <td>5494.0</td>
      <td>355700.0</td>
      <td>0.015211</td>
    </tr>
    <tr>
      <th>R07</th>
      <td>Diagnoses - main ICD10: R07 Pain in throat and...</td>
      <td>0.100279</td>
      <td>z7</td>
      <td>high</td>
      <td>24530.0</td>
      <td>336664.0</td>
      <td>0.067914</td>
    </tr>
    <tr>
      <th>M20</th>
      <td>Diagnoses - main ICD10: M20 Acquired deformiti...</td>
      <td>0.096417</td>
      <td>z4</td>
      <td>medium</td>
      <td>7773.0</td>
      <td>353421.0</td>
      <td>0.021520</td>
    </tr>
    <tr>
      <th>H26</th>
      <td>Diagnoses - main ICD10: H26 Other cataract</td>
      <td>0.086648</td>
      <td>z4</td>
      <td>high</td>
      <td>11306.0</td>
      <td>349888.0</td>
      <td>0.031302</td>
    </tr>
    <tr>
      <th>K21</th>
      <td>Diagnoses - main ICD10: K21 Gastro-oesophageal...</td>
      <td>0.081990</td>
      <td>z4</td>
      <td>high</td>
      <td>10743.0</td>
      <td>350451.0</td>
      <td>0.029743</td>
    </tr>
    <tr>
      <th>K44</th>
      <td>Diagnoses - main ICD10: K44 Diaphragmatic hernia</td>
      <td>0.075533</td>
      <td>z4</td>
      <td>medium</td>
      <td>8042.0</td>
      <td>353152.0</td>
      <td>0.022265</td>
    </tr>
    <tr>
      <th>M23</th>
      <td>Diagnoses - main ICD10: M23 Internal derangeme...</td>
      <td>0.074823</td>
      <td>z4</td>
      <td>high</td>
      <td>11831.0</td>
      <td>349363.0</td>
      <td>0.032755</td>
    </tr>
    <tr>
      <th>N39</th>
      <td>Diagnoses - main ICD10: N39 Other disorders of...</td>
      <td>0.063935</td>
      <td>z4</td>
      <td>high</td>
      <td>10551.0</td>
      <td>350643.0</td>
      <td>0.029211</td>
    </tr>
    <tr>
      <th>R10</th>
      <td>Diagnoses - main ICD10: R10 Abdominal and pelv...</td>
      <td>0.062420</td>
      <td>z4</td>
      <td>high</td>
      <td>20240.0</td>
      <td>340954.0</td>
      <td>0.056036</td>
    </tr>
    <tr>
      <th>R31</th>
      <td>Diagnoses - main ICD10: R31 Unspecified haemat...</td>
      <td>0.060724</td>
      <td>z4</td>
      <td>high</td>
      <td>11283.0</td>
      <td>349911.0</td>
      <td>0.031238</td>
    </tr>
    <tr>
      <th>I84</th>
      <td>Diagnoses - main ICD10: I84 Haemorrhoids</td>
      <td>0.057475</td>
      <td>z4</td>
      <td>high</td>
      <td>12102.0</td>
      <td>349092.0</td>
      <td>0.033506</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">index</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>Index([&#39;I83&#39;, &#39;I25&#39;, &#39;G56&#39;, &#39;N81&#39;, &#39;M16&#39;, &#39;M17&#39;, &#39;I48&#39;, &#39;C44&#39;, &#39;K80&#39;, &#39;I21&#39;,
       &#39;I20&#39;, &#39;K57&#39;, &#39;K40&#39;, &#39;D12&#39;, &#39;M54&#39;, &#39;C50&#39;, &#39;K22&#39;, &#39;R07&#39;, &#39;M20&#39;, &#39;H26&#39;,
       &#39;K21&#39;, &#39;K44&#39;, &#39;M23&#39;, &#39;N39&#39;, &#39;R10&#39;, &#39;R31&#39;, &#39;I84&#39;],
      dtype=&#39;object&#39;, name=&#39;phenotype&#39;)</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">shape</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">pheno</span> <span class="o">=</span> <span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s2">&quot;I84&quot;</span><span class="p">]</span>
<span class="n">pheno</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>description         Diagnoses - main ICD10: I84 Haemorrhoids
h2_liability                                        0.057475
h2_liability_se                                     0.013482
h2_observed                                         0.009864
h2_observed_se                                      0.002314
h2_z                                                4.263044
h2_p                                                 0.00001
h2_sig                                                    z4
confidence                                              high
notes                                                    NaN
intercept                                           1.018834
intercept_se                                        0.008843
intercept_z                                         2.129842
intercept_p                                         0.016592
lambdaGC                                            1.074945
mean_chi2                                           1.085477
ratio                                               0.220335
ratio_se                                            0.103451
n                                                     361194
Neff                                                46786.06
variable_type                                         binary
isBinary                                                True
n_cases                                              12102.0
n_controls                                          349092.0
prevalence                                          0.033506
source                                                 icd10
sex                                               both_sexes
isNotPrimary                                           False
isBadPower                                             False
isLowNeff                                              False
isMidNeff                                              False
isExtremeSE                                            False
isHighSE                                               False
isSexBias                                              False
isBadOrdinal                                           False
isNumericOrdinal                                       False
Name: I84, dtype: object</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Output-phenotype-matrix">Output phenotype matrix<a class="anchor-link" href="#Output-phenotype-matrix"> </a></h3>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="upsample_pheno" class="doc_header"><code>upsample_pheno</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/phenotypes/__init__.py#L70" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>upsample_pheno</code>(<strong><code>pheno_df</code></strong>, <strong><code>balance_pheno</code></strong>, <strong><code>max_samples</code></strong>, <strong><code>random_state</code></strong>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_phenotype" class="doc_header"><code>get_phenotype</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/phenotypes/__init__.py#L76" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_phenotype</code>(<strong><code>icd10_codes</code></strong>:<code>Union</code>[<code>str</code>, <code>List</code>[<code>str</code>]]=<em><code>'I84'</code></em>, <strong><code>samples</code></strong>:<code>array</code>=<em><code>None</code></em>, <strong><code>max_samples</code></strong>:<code>int</code>=<em><code>None</code></em>, <strong><code>balance_pheno</code></strong>:<code>str</code>=<em><code>None</code></em>, <strong><code>random_state</code></strong>=<em><code>42</code></em>)</p>
</blockquote>
<p>if samples argument is provided from genetic file, then find common set of samples and output ordered phenotype</p>
<p>if <code>max_samples</code> is provided, then over-sample the data so that we have <code>max_samples/2</code> for both cases and controls</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">test_one_pheno</span> <span class="o">=</span> <span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="s2">&quot;I84&quot;</span><span class="p">)</span>
<span class="n">test_one_pheno</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1000025</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000038</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000042</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000056</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000061</th>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
    </tr>
    <tr>
      <th>5873158</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873167</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873175</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873180</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873199</th>
      <td>0</td>
    </tr>
  </tbody>
</table>
<p>357682 rows Ã— 1 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="n">samples</span> <span class="o">=</span> <span class="n">sample_ids</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>5542886</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5137974</th>
      <td>0</td>
    </tr>
    <tr>
      <th>3758348</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1391800</th>
      <td>0</td>
    </tr>
    <tr>
      <th>3165331</th>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
    </tr>
    <tr>
      <th>3783812</th>
      <td>0</td>
    </tr>
    <tr>
      <th>2440570</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1213317</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1354423</th>
      <td>0</td>
    </tr>
    <tr>
      <th>4271450</th>
      <td>0</td>
    </tr>
  </tbody>
</table>
<p>357643 rows Ã— 1 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="n">samples</span> <span class="o">=</span> <span class="n">sample_ids</span><span class="p">)[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>I84
0      344300
1       13343
dtype: int64</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="n">samples</span> <span class="o">=</span> <span class="n">sample_ids</span><span class="p">)[</span><span class="mi">1</span><span class="p">][:</span><span class="mi">1000</span><span class="p">]</span><span class="o">.</span><span class="n">value_counts</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>I84
0      976
1       24
dtype: int64</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="s2">&quot;R07&quot;</span><span class="p">])[</span><span class="mi">1</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
      <th>R07</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1000025</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000038</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000042</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000056</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000061</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>5873158</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873167</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873175</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873180</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873199</th>
      <td>0</td>
      <td>0</td>
    </tr>
  </tbody>
</table>
<p>357682 rows Ã— 2 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Making-phenotype-balanced-in-case/control-number">Making phenotype balanced in case/control number<a class="anchor-link" href="#Making-phenotype-balanced-in-case/control-number"> </a></h3>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">test_phenotype_balanced</span> <span class="o">=</span>  <span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="s2">&quot;R07&quot;</span><span class="p">],</span> <span class="n">samples</span><span class="o">=</span><span class="n">sample_ids</span><span class="p">,</span> <span class="n">max_samples</span><span class="o">=</span><span class="mi">10000</span><span class="p">,</span> <span class="n">balance_pheno</span> <span class="o">=</span> <span class="s2">&quot;R07&quot;</span><span class="p">)</span>
<span class="n">test_phenotype_balanced</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
      <th>R07</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>5683283</th>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1309216</th>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>4200400</th>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>4615802</th>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>1487885</th>
      <td>0</td>
      <td>1</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>4143614</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2696228</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1929723</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>4827876</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>2509818</th>
      <td>1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
<p>10000 rows Ã— 2 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Only one phenotype is balanced, the others are not. The balanced phenotype is sampled with replacement with <code>weights = 1/[class frequency]</code></p>

</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">test_phenotype_balanced</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">Series</span><span class="o">.</span><span class="n">value_counts</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
      <th>R07</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>9494</td>
      <td>4926</td>
    </tr>
    <tr>
      <th>1</th>
      <td>506</td>
      <td>5074</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>
<h2 id="Get-SNPs-used-in-GWAS-for-trait">Get SNPs used in GWAS for trait<a class="anchor-link" href="#Get-SNPs-used-in-GWAS-for-trait"> </a></h2>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_GWAS_snps_for_trait" class="doc_header"><code>get_GWAS_snps_for_trait</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/phenotypes/__init__.py#L110" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_GWAS_snps_for_trait</code>(<strong><code>phenotype_code</code></strong>=<em><code>'I84'</code></em>, <strong><code>chromosome</code></strong>:<code>Union</code>[<code>int</code>, <code>List</code>[<code>int</code>]]=<em><code>21</code></em>, <strong><code>sort_val_cols_list</code></strong>:<code>List</code>[<code>str</code>]=<em><code>['pval']</code></em>, <strong><code>ascending_bool_list</code></strong>:<code>List</code>[<code>bool</code>]=<em><code>[True]</code></em>, <strong><code>id_only</code></strong>=<em><code>True</code></em>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_GWAS_snps_for_trait</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>array([&#39;21:46875774_C_T&#39;, &#39;21:30968890_C_T&#39;, &#39;21:43767602_A_G&#39;, ...,
       &#39;21:46876064_G_C&#39;, &#39;21:47773893_C_G&#39;, &#39;21:47831648_C_T&#39;],
      dtype=object)</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_GWAS_snps_for_trait</span><span class="p">(</span><span class="n">id_only</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>{21:         position_rank          variant minor_allele  minor_AF  \
 170594         170594  21:46875774:C:T            T  0.000007   
 84094           84094  21:30968890:C:T            T  0.000002   
 151278         151278  21:43767602:A:G            G  0.000002   
 88826           88826  21:31869421:T:G            G  0.000002   
 162176         162176  21:45587940:C:T            T  0.000003   
 ...               ...              ...          ...       ...   
 161896         161896  21:45545974:C:T            T  0.000000   
 165184         165184  21:46067181:G:C            C  0.000000   
 170601         170601  21:46876064:G:C            C  0.000000   
 176072         176072  21:47773893:C:G            G  0.000000   
 176489         176489  21:47831648:C:T            T  0.000000   
 
         expected_case_minor_AC  low_confidence_variant  n_complete_samples  \
 170594                0.172184                    True              361194   
 84094                 0.043356                    True              361194   
 151278                0.053403                    True              361194   
 88826                 0.041291                    True              361194   
 162176                0.068543                    True              361194   
 ...                        ...                     ...                 ...   
 161896                0.000000                    True              361194   
 165184                0.000000                    True              361194   
 170601                0.000000                    True              361194   
 176072                0.000000                    True              361194   
 176489                0.000000                    True              361194   
 
              AC       ytx      beta        se    tstat          pval chr  \
 170594  4.90588  2.117650  0.667492  0.107762  6.19411  5.867920e-10  21   
 84094   1.23529  1.000000  0.994117  0.186704  5.32456  1.012560e-07  21   
 151278  1.52157  0.831373  1.189730  0.226778  5.24623  1.553320e-07  21   
 88826   1.17647  0.952941  0.990411  0.192736  5.13868  2.768140e-07  21   
 162176  1.95294  0.933333  0.871227  0.184111  4.73207  2.223280e-06  21   
 ...         ...       ...       ...       ...      ...           ...  ..   
 161896  0.00000  0.000000       NaN       NaN      NaN           NaN  21   
 165184  0.00000  0.000000       NaN       NaN      NaN           NaN  21   
 170601  0.00000  0.000000       NaN       NaN      NaN           NaN  21   
 176072  0.00000  0.000000       NaN       NaN      NaN           NaN  21   
 176489  0.00000  0.000000       NaN       NaN      NaN           NaN  21   
 
         position major_allele          full_id  
 170594  46875774            C  21:46875774_C_T  
 84094   30968890            C  21:30968890_C_T  
 151278  43767602            A  21:43767602_A_G  
 88826   31869421            T  21:31869421_T_G  
 162176  45587940            C  21:45587940_C_T  
 ...          ...          ...              ...  
 161896  45545974            C  21:45545974_C_T  
 165184  46067181            G  21:46067181_G_C  
 170601  46876064            G  21:46876064_G_C  
 176072  47773893            C  21:47773893_C_G  
 176489  47831648            C  21:47831648_C_T  
 
 [178155 rows x 17 columns]}</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">test</span> <span class="o">=</span> <span class="n">get_GWAS_snps_for_trait</span><span class="p">(</span><span class="n">chromosome</span> <span class="o">=</span> <span class="p">[</span><span class="mi">20</span><span class="p">,</span> <span class="mi">21</span><span class="p">],</span> <span class="n">id_only</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">test</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>position_rank</th>
      <th>variant</th>
      <th>minor_allele</th>
      <th>minor_AF</th>
      <th>expected_case_minor_AC</th>
      <th>low_confidence_variant</th>
      <th>n_complete_samples</th>
      <th>AC</th>
      <th>ytx</th>
      <th>beta</th>
      <th>se</th>
      <th>tstat</th>
      <th>pval</th>
      <th>chr</th>
      <th>position</th>
      <th>major_allele</th>
      <th>full_id</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>466355</th>
      <td>466355</td>
      <td>21:46875774:C:T</td>
      <td>T</td>
      <td>6.791200e-06</td>
      <td>0.172184</td>
      <td>True</td>
      <td>361194</td>
      <td>4.905880</td>
      <td>2.117650</td>
      <td>0.667492</td>
      <td>0.107762</td>
      <td>6.19411</td>
      <td>5.867920e-10</td>
      <td>21</td>
      <td>46875774</td>
      <td>C</td>
      <td>21:46875774_C_T</td>
    </tr>
    <tr>
      <th>194304</th>
      <td>194304</td>
      <td>20:43850800:C:T</td>
      <td>T</td>
      <td>1.737160e-07</td>
      <td>0.004404</td>
      <td>True</td>
      <td>361194</td>
      <td>0.125490</td>
      <td>0.086275</td>
      <td>12.826500</td>
      <td>2.294580</td>
      <td>5.58991</td>
      <td>2.273570e-08</td>
      <td>20</td>
      <td>43850800</td>
      <td>C</td>
      <td>20:43850800_C_T</td>
    </tr>
    <tr>
      <th>149090</th>
      <td>149090</td>
      <td>20:33444611:C:T</td>
      <td>T</td>
      <td>1.644870e-06</td>
      <td>0.041704</td>
      <td>True</td>
      <td>361194</td>
      <td>1.188240</td>
      <td>0.568627</td>
      <td>2.003390</td>
      <td>0.359746</td>
      <td>5.56892</td>
      <td>2.565050e-08</td>
      <td>20</td>
      <td>33444611</td>
      <td>C</td>
      <td>20:33444611_C_T</td>
    </tr>
    <tr>
      <th>83527</th>
      <td>83527</td>
      <td>20:16253954:A:G</td>
      <td>G</td>
      <td>8.088640e-07</td>
      <td>0.020508</td>
      <td>True</td>
      <td>361194</td>
      <td>0.584314</td>
      <td>0.533333</td>
      <td>1.910680</td>
      <td>0.354733</td>
      <td>5.38626</td>
      <td>7.198550e-08</td>
      <td>20</td>
      <td>16253954</td>
      <td>A</td>
      <td>20:16253954_A_G</td>
    </tr>
    <tr>
      <th>183263</th>
      <td>183263</td>
      <td>20:41514559:A:G</td>
      <td>G</td>
      <td>3.257170e-08</td>
      <td>0.000826</td>
      <td>True</td>
      <td>361194</td>
      <td>0.023529</td>
      <td>0.011765</td>
      <td>88.836800</td>
      <td>16.586000</td>
      <td>5.35613</td>
      <td>8.507440e-08</td>
      <td>20</td>
      <td>41514559</td>
      <td>A</td>
      <td>20:41514559_A_G</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>457657</th>
      <td>457657</td>
      <td>21:45545974:C:T</td>
      <td>T</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>45545974</td>
      <td>C</td>
      <td>21:45545974_C_T</td>
    </tr>
    <tr>
      <th>460945</th>
      <td>460945</td>
      <td>21:46067181:G:C</td>
      <td>C</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46067181</td>
      <td>G</td>
      <td>21:46067181_G_C</td>
    </tr>
    <tr>
      <th>466362</th>
      <td>466362</td>
      <td>21:46876064:G:C</td>
      <td>C</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46876064</td>
      <td>G</td>
      <td>21:46876064_G_C</td>
    </tr>
    <tr>
      <th>471833</th>
      <td>471833</td>
      <td>21:47773893:C:G</td>
      <td>G</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47773893</td>
      <td>C</td>
      <td>21:47773893_C_G</td>
    </tr>
    <tr>
      <th>472250</th>
      <td>472250</td>
      <td>21:47831648:C:T</td>
      <td>T</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47831648</td>
      <td>C</td>
      <td>21:47831648_C_T</td>
    </tr>
  </tbody>
</table>
<p>473916 rows Ã— 17 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">assert</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">test</span><span class="p">[</span><span class="s2">&quot;chr&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">==</span> <span class="p">[</span><span class="s1">&#39;20&#39;</span><span class="p">,</span> <span class="s1">&#39;21&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

</div>
 

