---

title: Functions to extract phenotype info


keywords: fastai
sidebar: home_sidebar

summary: "API details."
description: "API details."
nb_path: "notebooks/package/phenotypes.ipynb"
---
<!--

#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: notebooks/package/phenotypes.ipynb
# command to build the docs after a change: nbdev_build_docs

-->

<div class="container" id="notebook-container">
        
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/combinatorial_gwas/phenotypes/__init__.py:22: UserWarning: registration of accessor &lt;class &#39;combinatorial_gwas.phenotypes.__init__.QueryDataframe&#39;&gt; under name &#39;pheno&#39; for type &lt;class &#39;pandas.core.frame.DataFrame&#39;&gt; is overriding a preexisting attribute with the same name.
  class QueryDataframe():
WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h2 id="QueryDataframe" class="doc_header"><code>class</code> <code>QueryDataframe</code><a href="https://github.com/hoangthienan95/combinatorial_gwas/tree/master/combinatorial_gwas/phenotypes/__init__.py#L22" class="source_link" style="float:right">[source]</a></h2><blockquote><p><code>QueryDataframe</code>(<strong><code>df</code></strong>:<code>DataFrame</code>)</p>
</blockquote>
<p>QueryDataframe(df: pandas.core.frame.DataFrame)</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>&lt;ipython-input-21-dcb25e435725&gt;:5: UserWarning: registration of accessor &lt;class &#39;__main__.QueryDataframe&#39;&gt; under name &#39;pheno&#39; for type &lt;class &#39;pandas.core.frame.DataFrame&#39;&gt; is overriding a preexisting attribute with the same name.
  class QueryDataframe():
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>{&#39;example_test_data_ratio&#39;: 0.2,
 &#39;example_num_train_iter&#39;: 10000,
 &#39;example_learning_rate&#39;: 0.01,
 &#39;template_gwas_result_file_link&#39;: &#39;https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/{phenotype_code}.gwas.imputed_v3.both_sexes.tsv.bgz&#39;,
 &#39;genetic_file_path_template&#39;: &#39;/lab/corradin_biobank/Raw_UKB_downloads/BGEN/ukb_imp_chr{chrom_number}_v3.bgen&#39;,
 &#39;sample_file_template&#39;: &#39;/lab/corradin_biobank/Raw_UKB_downloads/sample_files/ukb45624_imp_chr21_v3_s487275.sample&#39;,
 &#39;data_dict_file&#39;: &#39;/lab/corradin_biobank/Phenotypes//Data_Dictionary_Showcase.csv&#39;,
 &#39;pheno_file&#39;: &#39;/lab/corradin_biobank/samples//neale_gwas_both_sexes_parquet/&#39;,
 &#39;coding_file_path_template&#39;: &#39;/lab/corradin_biobank/Phenotypes//data_codes/datacode-*.tsv&#39;,
 &#39;id_col&#39;: &#39;f.eid&#39;}</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>[&#39;example_iris_data&#39;,
 &#39;globals_test&#39;,
 &#39;disease_summary&#39;,
 &#39;ICD10_pheno_matrix&#39;,
 &#39;phenotypes_info_Neale&#39;,
 &#39;GWAS_results_links_df&#39;,
 &#39;heritability_trait_level_summary&#39;,
 &#39;heritability_trait_level_partitioned_z4_sig&#39;]</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Find-high-heritability-traits">Find high heritability traits<a class="anchor-link" href="#Find-high-heritability-traits"> </a></h3>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">heritability_Neale</span> <span class="o">=</span> <span class="n">catalog_all</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="s2">&quot;heritability_trait_level_summary&quot;</span><span class="p">)</span>
<span class="n">heritability_Neale</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/extras/datasets/pandas/csv_dataset.py:158: ParserWarning: Falling back to the &#39;python&#39; engine because the &#39;c&#39; engine does not support regex separators (separators &gt; 1 char and different from &#39;\s+&#39; are interpreted as regex); you can avoid this warning by specifying engine=&#39;python&#39;.
  return pd.read_csv(fs_file, **self._load_args)
</pre>
</div>
</div>

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>phenotype</th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_liability_se</th>
      <th>h2_observed</th>
      <th>h2_observed_se</th>
      <th>h2_z</th>
      <th>h2_p</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>...</th>
      <th>sex</th>
      <th>isNotPrimary</th>
      <th>isBadPower</th>
      <th>isLowNeff</th>
      <th>isMidNeff</th>
      <th>isExtremeSE</th>
      <th>isHighSE</th>
      <th>isSexBias</th>
      <th>isBadOrdinal</th>
      <th>isNumericOrdinal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100001_irnt</td>
      <td>Food weight</td>
      <td>0.068818</td>
      <td>0.016857</td>
      <td>0.068818</td>
      <td>0.016857</td>
      <td>4.082528</td>
      <td>0.000022</td>
      <td>z4</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100002_irnt</td>
      <td>Energy</td>
      <td>0.064783</td>
      <td>0.015784</td>
      <td>0.064783</td>
      <td>0.015784</td>
      <td>4.104249</td>
      <td>0.000020</td>
      <td>z4</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100003_irnt</td>
      <td>Protein</td>
      <td>0.034543</td>
      <td>0.014619</td>
      <td>0.034543</td>
      <td>0.014619</td>
      <td>2.362892</td>
      <td>0.009066</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>3</th>
      <td>100004_irnt</td>
      <td>Fat</td>
      <td>0.055197</td>
      <td>0.018265</td>
      <td>0.055197</td>
      <td>0.018265</td>
      <td>3.021937</td>
      <td>0.001256</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
    <tr>
      <th>4</th>
      <td>100005_irnt</td>
      <td>Carbohydrate</td>
      <td>0.051744</td>
      <td>0.015357</td>
      <td>0.051744</td>
      <td>0.015357</td>
      <td>3.369359</td>
      <td>0.000377</td>
      <td>nominal</td>
      <td>high</td>
      <td>...</td>
      <td>both_sexes</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
      <td>False</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 37 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">heritability_Neale</span><span class="o">.</span><span class="n">columns</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>Index([&#39;phenotype&#39;, &#39;description&#39;, &#39;h2_liability&#39;, &#39;h2_liability_se&#39;,
       &#39;h2_observed&#39;, &#39;h2_observed_se&#39;, &#39;h2_z&#39;, &#39;h2_p&#39;, &#39;h2_sig&#39;, &#39;confidence&#39;,
       &#39;notes&#39;, &#39;intercept&#39;, &#39;intercept_se&#39;, &#39;intercept_z&#39;, &#39;intercept_p&#39;,
       &#39;lambdaGC&#39;, &#39;mean_chi2&#39;, &#39;ratio&#39;, &#39;ratio_se&#39;, &#39;n&#39;, &#39;Neff&#39;,
       &#39;variable_type&#39;, &#39;isBinary&#39;, &#39;n_cases&#39;, &#39;n_controls&#39;, &#39;prevalence&#39;,
       &#39;source&#39;, &#39;sex&#39;, &#39;isNotPrimary&#39;, &#39;isBadPower&#39;, &#39;isLowNeff&#39;, &#39;isMidNeff&#39;,
       &#39;isExtremeSE&#39;, &#39;isHighSE&#39;, &#39;isSexBias&#39;, &#39;isBadOrdinal&#39;,
       &#39;isNumericOrdinal&#39;],
      dtype=&#39;object&#39;)</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">display_cols</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;description&#39;</span><span class="p">,</span> <span class="s1">&#39;h2_liability&#39;</span><span class="p">,</span> <span class="s1">&#39;h2_sig&#39;</span><span class="p">,</span> <span class="s1">&#39;confidence&#39;</span><span class="p">,</span> <span class="s1">&#39;n_cases&#39;</span><span class="p">,</span> <span class="s1">&#39;n_controls&#39;</span><span class="p">,</span> <span class="s1">&#39;prevalence&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span> <span class="o">=</span> <span class="n">heritability_Neale</span><span class="o">.</span><span class="n">pheno</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="n">h2_sig</span> <span class="o">=</span> <span class="s2">&quot;in [&#39;z7&#39;, &#39;z4&#39;]&quot;</span><span class="p">,</span> <span class="n">source</span><span class="o">=</span> <span class="s2">&quot; == &#39;icd10&#39;&quot;</span><span class="p">,</span> <span class="n">confidence</span><span class="o">=</span> <span class="s2">&quot;in [&#39;medium&#39;, &#39;high&#39;]&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s2">&quot;h2_liability&quot;</span><span class="p">,</span> <span class="n">ascending</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
<span class="n">quality_heritability_phenos</span> <span class="o">=</span> <span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;phenotype&quot;</span><span class="p">)</span>
<span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">head</span><span class="p">()[</span><span class="n">display_cols</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>n_cases</th>
      <th>n_controls</th>
      <th>prevalence</th>
    </tr>
    <tr>
      <th>phenotype</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>I83</th>
      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>
      <td>0.189299</td>
      <td>z7</td>
      <td>medium</td>
      <td>8763.0</td>
      <td>352431.0</td>
      <td>0.024261</td>
    </tr>
    <tr>
      <th>I25</th>
      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>
      <td>0.163823</td>
      <td>z7</td>
      <td>medium</td>
      <td>12769.0</td>
      <td>348425.0</td>
      <td>0.035352</td>
    </tr>
    <tr>
      <th>G56</th>
      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>
      <td>0.157519</td>
      <td>z7</td>
      <td>medium</td>
      <td>8130.0</td>
      <td>353064.0</td>
      <td>0.022509</td>
    </tr>
    <tr>
      <th>N81</th>
      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>
      <td>0.152223</td>
      <td>z4</td>
      <td>medium</td>
      <td>7511.0</td>
      <td>186663.0</td>
      <td>0.038682</td>
    </tr>
    <tr>
      <th>M16</th>
      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>
      <td>0.151297</td>
      <td>z4</td>
      <td>medium</td>
      <td>9136.0</td>
      <td>352058.0</td>
      <td>0.025294</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span><span class="p">[</span><span class="n">display_cols</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>description</th>
      <th>h2_liability</th>
      <th>h2_sig</th>
      <th>confidence</th>
      <th>n_cases</th>
      <th>n_controls</th>
      <th>prevalence</th>
    </tr>
    <tr>
      <th>phenotype</th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>I83</th>
      <td>Diagnoses - main ICD10: I83 Varicose veins of ...</td>
      <td>0.189299</td>
      <td>z7</td>
      <td>medium</td>
      <td>8763.0</td>
      <td>352431.0</td>
      <td>0.024261</td>
    </tr>
    <tr>
      <th>I25</th>
      <td>Diagnoses - main ICD10: I25 Chronic ischaemic ...</td>
      <td>0.163823</td>
      <td>z7</td>
      <td>medium</td>
      <td>12769.0</td>
      <td>348425.0</td>
      <td>0.035352</td>
    </tr>
    <tr>
      <th>G56</th>
      <td>Diagnoses - main ICD10: G56 Mononeuropathies o...</td>
      <td>0.157519</td>
      <td>z7</td>
      <td>medium</td>
      <td>8130.0</td>
      <td>353064.0</td>
      <td>0.022509</td>
    </tr>
    <tr>
      <th>N81</th>
      <td>Diagnoses - main ICD10: N81 Female genital pro...</td>
      <td>0.152223</td>
      <td>z4</td>
      <td>medium</td>
      <td>7511.0</td>
      <td>186663.0</td>
      <td>0.038682</td>
    </tr>
    <tr>
      <th>M16</th>
      <td>Diagnoses - main ICD10: M16 Coxarthrosis [arth...</td>
      <td>0.151297</td>
      <td>z4</td>
      <td>medium</td>
      <td>9136.0</td>
      <td>352058.0</td>
      <td>0.025294</td>
    </tr>
    <tr>
      <th>M17</th>
      <td>Diagnoses - main ICD10: M17 Gonarthrosis [arth...</td>
      <td>0.144360</td>
      <td>z7</td>
      <td>high</td>
      <td>11497.0</td>
      <td>349697.0</td>
      <td>0.031831</td>
    </tr>
    <tr>
      <th>I48</th>
      <td>Diagnoses - main ICD10: I48 Atrial fibrillatio...</td>
      <td>0.144326</td>
      <td>z4</td>
      <td>medium</td>
      <td>6356.0</td>
      <td>354838.0</td>
      <td>0.017597</td>
    </tr>
    <tr>
      <th>C44</th>
      <td>Diagnoses - main ICD10: C44 Other malignant ne...</td>
      <td>0.141495</td>
      <td>z4</td>
      <td>medium</td>
      <td>9086.0</td>
      <td>352108.0</td>
      <td>0.025155</td>
    </tr>
    <tr>
      <th>K80</th>
      <td>Diagnoses - main ICD10: K80 Cholelithiasis</td>
      <td>0.137236</td>
      <td>z4</td>
      <td>high</td>
      <td>10520.0</td>
      <td>350674.0</td>
      <td>0.029126</td>
    </tr>
    <tr>
      <th>I21</th>
      <td>Diagnoses - main ICD10: I21 Acute myocardial i...</td>
      <td>0.136628</td>
      <td>z4</td>
      <td>medium</td>
      <td>5948.0</td>
      <td>355246.0</td>
      <td>0.016468</td>
    </tr>
    <tr>
      <th>I20</th>
      <td>Diagnoses - main ICD10: I20 Angina pectoris</td>
      <td>0.129187</td>
      <td>z4</td>
      <td>medium</td>
      <td>6246.0</td>
      <td>354948.0</td>
      <td>0.017293</td>
    </tr>
    <tr>
      <th>K57</th>
      <td>Diagnoses - main ICD10: K57 Diverticular disea...</td>
      <td>0.129184</td>
      <td>z7</td>
      <td>high</td>
      <td>12662.0</td>
      <td>348532.0</td>
      <td>0.035056</td>
    </tr>
    <tr>
      <th>K40</th>
      <td>Diagnoses - main ICD10: K40 Inguinal hernia</td>
      <td>0.120213</td>
      <td>z7</td>
      <td>medium</td>
      <td>13147.0</td>
      <td>348047.0</td>
      <td>0.036399</td>
    </tr>
    <tr>
      <th>D12</th>
      <td>Diagnoses - main ICD10: D12 Benign neoplasm of...</td>
      <td>0.118093</td>
      <td>z4</td>
      <td>medium</td>
      <td>8877.0</td>
      <td>352317.0</td>
      <td>0.024577</td>
    </tr>
    <tr>
      <th>M54</th>
      <td>Diagnoses - main ICD10: M54 Dorsalgia</td>
      <td>0.114167</td>
      <td>z7</td>
      <td>medium</td>
      <td>8361.0</td>
      <td>352833.0</td>
      <td>0.023148</td>
    </tr>
    <tr>
      <th>C50</th>
      <td>Diagnoses - main ICD10: C50 Malignant neoplasm...</td>
      <td>0.110280</td>
      <td>z4</td>
      <td>medium</td>
      <td>8304.0</td>
      <td>352890.0</td>
      <td>0.022990</td>
    </tr>
    <tr>
      <th>K22</th>
      <td>Diagnoses - main ICD10: K22 Other diseases of ...</td>
      <td>0.107578</td>
      <td>z4</td>
      <td>medium</td>
      <td>5494.0</td>
      <td>355700.0</td>
      <td>0.015211</td>
    </tr>
    <tr>
      <th>R07</th>
      <td>Diagnoses - main ICD10: R07 Pain in throat and...</td>
      <td>0.100279</td>
      <td>z7</td>
      <td>high</td>
      <td>24530.0</td>
      <td>336664.0</td>
      <td>0.067914</td>
    </tr>
    <tr>
      <th>M20</th>
      <td>Diagnoses - main ICD10: M20 Acquired deformiti...</td>
      <td>0.096417</td>
      <td>z4</td>
      <td>medium</td>
      <td>7773.0</td>
      <td>353421.0</td>
      <td>0.021520</td>
    </tr>
    <tr>
      <th>H26</th>
      <td>Diagnoses - main ICD10: H26 Other cataract</td>
      <td>0.086648</td>
      <td>z4</td>
      <td>high</td>
      <td>11306.0</td>
      <td>349888.0</td>
      <td>0.031302</td>
    </tr>
    <tr>
      <th>K21</th>
      <td>Diagnoses - main ICD10: K21 Gastro-oesophageal...</td>
      <td>0.081990</td>
      <td>z4</td>
      <td>high</td>
      <td>10743.0</td>
      <td>350451.0</td>
      <td>0.029743</td>
    </tr>
    <tr>
      <th>K44</th>
      <td>Diagnoses - main ICD10: K44 Diaphragmatic hernia</td>
      <td>0.075533</td>
      <td>z4</td>
      <td>medium</td>
      <td>8042.0</td>
      <td>353152.0</td>
      <td>0.022265</td>
    </tr>
    <tr>
      <th>M23</th>
      <td>Diagnoses - main ICD10: M23 Internal derangeme...</td>
      <td>0.074823</td>
      <td>z4</td>
      <td>high</td>
      <td>11831.0</td>
      <td>349363.0</td>
      <td>0.032755</td>
    </tr>
    <tr>
      <th>N39</th>
      <td>Diagnoses - main ICD10: N39 Other disorders of...</td>
      <td>0.063935</td>
      <td>z4</td>
      <td>high</td>
      <td>10551.0</td>
      <td>350643.0</td>
      <td>0.029211</td>
    </tr>
    <tr>
      <th>R10</th>
      <td>Diagnoses - main ICD10: R10 Abdominal and pelv...</td>
      <td>0.062420</td>
      <td>z4</td>
      <td>high</td>
      <td>20240.0</td>
      <td>340954.0</td>
      <td>0.056036</td>
    </tr>
    <tr>
      <th>R31</th>
      <td>Diagnoses - main ICD10: R31 Unspecified haemat...</td>
      <td>0.060724</td>
      <td>z4</td>
      <td>high</td>
      <td>11283.0</td>
      <td>349911.0</td>
      <td>0.031238</td>
    </tr>
    <tr>
      <th>I84</th>
      <td>Diagnoses - main ICD10: I84 Haemorrhoids</td>
      <td>0.057475</td>
      <td>z4</td>
      <td>high</td>
      <td>12102.0</td>
      <td>349092.0</td>
      <td>0.033506</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">shape</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">pheno</span> <span class="o">=</span> <span class="n">quality_heritability_phenos</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s2">&quot;I84&quot;</span><span class="p">]</span>
<span class="n">pheno</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>description         Diagnoses - main ICD10: I84 Haemorrhoids
h2_liability                                        0.057475
h2_liability_se                                     0.013482
h2_observed                                         0.009864
h2_observed_se                                      0.002314
h2_z                                                4.263044
h2_p                                                 0.00001
h2_sig                                                    z4
confidence                                              high
notes                                                    NaN
intercept                                           1.018834
intercept_se                                        0.008843
intercept_z                                         2.129842
intercept_p                                         0.016592
lambdaGC                                            1.074945
mean_chi2                                           1.085477
ratio                                               0.220335
ratio_se                                            0.103451
n                                                     361194
Neff                                                46786.06
variable_type                                         binary
isBinary                                                True
n_cases                                              12102.0
n_controls                                          349092.0
prevalence                                          0.033506
source                                                 icd10
sex                                               both_sexes
isNotPrimary                                           False
isBadPower                                             False
isLowNeff                                              False
isMidNeff                                              False
isExtremeSE                                            False
isHighSE                                               False
isSexBias                                              False
isBadOrdinal                                           False
isNumericOrdinal                                       False
Name: I84, dtype: object</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Output-phenotype-matrix">Output phenotype matrix<a class="anchor-link" href="#Output-phenotype-matrix"> </a></h3>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>/Users/ahoang/Documents/MIT_classes/DeepLearning_Spring21/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/io/data_catalog.py:402: DtypeWarning: Columns (28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236) have mixed types.Specify dtype option on import or set low_memory=False.
  result = func()
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_phenotype" class="doc_header"><code>get_phenotype</code><a href="https://github.com/hoangthienan95/combinatorial_gwas/tree/master/combinatorial_gwas/phenotypes/__init__.py#L55" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_phenotype</code>(<strong><code>icd10_codes</code></strong>:<code>Union</code>[<code>str</code>, <code>List</code>[<code>str</code>]]=<em><code>'I84'</code></em>, <strong><code>samples</code></strong>:<code>Series</code>=<em><code>None</code></em>)</p>
</blockquote>
<p>if samples argument is provided from genetic file, then find common set of samples and output ordered phenotype</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="s2">&quot;I84&quot;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1000025</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000038</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000042</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000056</th>
      <td>0</td>
    </tr>
    <tr>
      <th>1000061</th>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
    </tr>
    <tr>
      <th>5873158</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873167</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873175</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873180</th>
      <td>0</td>
    </tr>
    <tr>
      <th>5873199</th>
      <td>0</td>
    </tr>
  </tbody>
</table>
<p>357682 rows × 1 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_phenotype</span><span class="p">(</span><span class="n">icd10_codes</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;I84&quot;</span><span class="p">,</span> <span class="s2">&quot;R07&quot;</span><span class="p">])</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>I84</th>
      <th>R07</th>
    </tr>
    <tr>
      <th>f.eid</th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1000025</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000038</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000042</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000056</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>1000061</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>5873158</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873167</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873175</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873180</th>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <th>5873199</th>
      <td>0</td>
      <td>0</td>
    </tr>
  </tbody>
</table>
<p>357682 rows × 2 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

<div class="cell border-box-sizing text_cell rendered"><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<hr>
<h2 id="Get-SNPs-used-in-GWAS-for-trait">Get SNPs used in GWAS for trait<a class="anchor-link" href="#Get-SNPs-used-in-GWAS-for-trait"> </a></h2>
</div>
</div>
</div>
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_GWAS_snps_for_trait" class="doc_header"><code>get_GWAS_snps_for_trait</code><a href="https://github.com/hoangthienan95/combinatorial_gwas/tree/master/combinatorial_gwas/phenotypes/__init__.py#L75" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_GWAS_snps_for_trait</code>(<strong><code>phenotype_code</code></strong>=<em><code>'I84'</code></em>, <strong><code>chromosome</code></strong>:<code>Union</code>[<code>int</code>, <code>List</code>[<code>int</code>]]=<em><code>21</code></em>, <strong><code>sort_val_cols_list</code></strong>:<code>List</code>[<code>str</code>]=<em><code>['pval']</code></em>, <strong><code>ascending_bool_list</code></strong>:<code>List</code>[<code>bool</code>]=<em><code>[True]</code></em>, <strong><code>id_only</code></strong>=<em><code>True</code></em>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_GWAS_snps_for_trait</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>array([&#39;21:46875774_C_T&#39;, &#39;21:30968890_C_T&#39;, &#39;21:43767602_A_G&#39;, ...,
       &#39;21:46876064_G_C&#39;, &#39;21:47773893_C_G&#39;, &#39;21:47831648_C_T&#39;],
      dtype=object)</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">get_GWAS_snps_for_trait</span><span class="p">(</span><span class="n">id_only</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>position_rank</th>
      <th>variant</th>
      <th>minor_allele</th>
      <th>minor_AF</th>
      <th>expected_case_minor_AC</th>
      <th>low_confidence_variant</th>
      <th>n_complete_samples</th>
      <th>AC</th>
      <th>ytx</th>
      <th>beta</th>
      <th>se</th>
      <th>tstat</th>
      <th>pval</th>
      <th>chr</th>
      <th>position</th>
      <th>major_allele</th>
      <th>full_id</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>46739</th>
      <td>46739</td>
      <td>21:23683958:G:A</td>
      <td>A</td>
      <td>0.448599</td>
      <td>11373.8000</td>
      <td>False</td>
      <td>361194</td>
      <td>324062.000</td>
      <td>11381.3000</td>
      <td>1.745250e-09</td>
      <td>0.000435</td>
      <td>0.000004</td>
      <td>0.999997</td>
      <td>21</td>
      <td>23683958</td>
      <td>G</td>
      <td>21:23683958_G_A</td>
    </tr>
    <tr>
      <th>156679</th>
      <td>156679</td>
      <td>21:44501642:C:T</td>
      <td>T</td>
      <td>0.012528</td>
      <td>317.6430</td>
      <td>False</td>
      <td>361194</td>
      <td>9050.310</td>
      <td>315.2900</td>
      <td>-1.938640e-08</td>
      <td>0.002030</td>
      <td>-0.000010</td>
      <td>0.999992</td>
      <td>21</td>
      <td>44501642</td>
      <td>C</td>
      <td>21:44501642_C_T</td>
    </tr>
    <tr>
      <th>57576</th>
      <td>57576</td>
      <td>21:25570916:T:C</td>
      <td>C</td>
      <td>0.006228</td>
      <td>157.8990</td>
      <td>False</td>
      <td>361194</td>
      <td>4498.860</td>
      <td>158.5530</td>
      <td>-5.141850e-08</td>
      <td>0.002875</td>
      <td>-0.000018</td>
      <td>0.999986</td>
      <td>21</td>
      <td>25570916</td>
      <td>T</td>
      <td>21:25570916_T_C</td>
    </tr>
    <tr>
      <th>17572</th>
      <td>17572</td>
      <td>21:18598893:A:C</td>
      <td>C</td>
      <td>0.001108</td>
      <td>28.0995</td>
      <td>False</td>
      <td>361194</td>
      <td>800.612</td>
      <td>28.0941</td>
      <td>1.672180e-07</td>
      <td>0.006937</td>
      <td>0.000024</td>
      <td>0.999981</td>
      <td>21</td>
      <td>18598893</td>
      <td>A</td>
      <td>21:18598893_A_C</td>
    </tr>
    <tr>
      <th>89085</th>
      <td>89085</td>
      <td>21:31934274:T:C</td>
      <td>C</td>
      <td>0.011455</td>
      <td>290.4320</td>
      <td>False</td>
      <td>361194</td>
      <td>8275.000</td>
      <td>288.0000</td>
      <td>7.702210e-08</td>
      <td>0.002036</td>
      <td>0.000038</td>
      <td>0.999970</td>
      <td>21</td>
      <td>31934274</td>
      <td>T</td>
      <td>21:31934274_T_C</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>161896</th>
      <td>161896</td>
      <td>21:45545974:C:T</td>
      <td>T</td>
      <td>0.000000</td>
      <td>0.0000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000</td>
      <td>0.0000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>45545974</td>
      <td>C</td>
      <td>21:45545974_C_T</td>
    </tr>
    <tr>
      <th>165184</th>
      <td>165184</td>
      <td>21:46067181:G:C</td>
      <td>C</td>
      <td>0.000000</td>
      <td>0.0000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000</td>
      <td>0.0000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46067181</td>
      <td>G</td>
      <td>21:46067181_G_C</td>
    </tr>
    <tr>
      <th>170601</th>
      <td>170601</td>
      <td>21:46876064:G:C</td>
      <td>C</td>
      <td>0.000000</td>
      <td>0.0000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000</td>
      <td>0.0000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46876064</td>
      <td>G</td>
      <td>21:46876064_G_C</td>
    </tr>
    <tr>
      <th>176072</th>
      <td>176072</td>
      <td>21:47773893:C:G</td>
      <td>G</td>
      <td>0.000000</td>
      <td>0.0000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000</td>
      <td>0.0000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47773893</td>
      <td>C</td>
      <td>21:47773893_C_G</td>
    </tr>
    <tr>
      <th>176489</th>
      <td>176489</td>
      <td>21:47831648:C:T</td>
      <td>T</td>
      <td>0.000000</td>
      <td>0.0000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000</td>
      <td>0.0000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47831648</td>
      <td>C</td>
      <td>21:47831648_C_T</td>
    </tr>
  </tbody>
</table>
<p>178155 rows × 17 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">test</span> <span class="o">=</span> <span class="n">get_GWAS_snps_for_trait</span><span class="p">(</span><span class="n">chromosome</span> <span class="o">=</span> <span class="p">[</span><span class="mi">20</span><span class="p">,</span> <span class="mi">21</span><span class="p">],</span> <span class="n">id_only</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">test</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>position_rank</th>
      <th>variant</th>
      <th>minor_allele</th>
      <th>minor_AF</th>
      <th>expected_case_minor_AC</th>
      <th>low_confidence_variant</th>
      <th>n_complete_samples</th>
      <th>AC</th>
      <th>ytx</th>
      <th>beta</th>
      <th>se</th>
      <th>tstat</th>
      <th>pval</th>
      <th>chr</th>
      <th>position</th>
      <th>major_allele</th>
      <th>full_id</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>466355</th>
      <td>466355</td>
      <td>21:46875774:C:T</td>
      <td>T</td>
      <td>6.791200e-06</td>
      <td>0.172184</td>
      <td>True</td>
      <td>361194</td>
      <td>4.905880</td>
      <td>2.117650</td>
      <td>0.667492</td>
      <td>0.107762</td>
      <td>6.19411</td>
      <td>5.867920e-10</td>
      <td>21</td>
      <td>46875774</td>
      <td>C</td>
      <td>21:46875774_C_T</td>
    </tr>
    <tr>
      <th>194304</th>
      <td>194304</td>
      <td>20:43850800:C:T</td>
      <td>T</td>
      <td>1.737160e-07</td>
      <td>0.004404</td>
      <td>True</td>
      <td>361194</td>
      <td>0.125490</td>
      <td>0.086275</td>
      <td>12.826500</td>
      <td>2.294580</td>
      <td>5.58991</td>
      <td>2.273570e-08</td>
      <td>20</td>
      <td>43850800</td>
      <td>C</td>
      <td>20:43850800_C_T</td>
    </tr>
    <tr>
      <th>149090</th>
      <td>149090</td>
      <td>20:33444611:C:T</td>
      <td>T</td>
      <td>1.644870e-06</td>
      <td>0.041704</td>
      <td>True</td>
      <td>361194</td>
      <td>1.188240</td>
      <td>0.568627</td>
      <td>2.003390</td>
      <td>0.359746</td>
      <td>5.56892</td>
      <td>2.565050e-08</td>
      <td>20</td>
      <td>33444611</td>
      <td>C</td>
      <td>20:33444611_C_T</td>
    </tr>
    <tr>
      <th>83527</th>
      <td>83527</td>
      <td>20:16253954:A:G</td>
      <td>G</td>
      <td>8.088640e-07</td>
      <td>0.020508</td>
      <td>True</td>
      <td>361194</td>
      <td>0.584314</td>
      <td>0.533333</td>
      <td>1.910680</td>
      <td>0.354733</td>
      <td>5.38626</td>
      <td>7.198550e-08</td>
      <td>20</td>
      <td>16253954</td>
      <td>A</td>
      <td>20:16253954_A_G</td>
    </tr>
    <tr>
      <th>183263</th>
      <td>183263</td>
      <td>20:41514559:A:G</td>
      <td>G</td>
      <td>3.257170e-08</td>
      <td>0.000826</td>
      <td>True</td>
      <td>361194</td>
      <td>0.023529</td>
      <td>0.011765</td>
      <td>88.836800</td>
      <td>16.586000</td>
      <td>5.35613</td>
      <td>8.507440e-08</td>
      <td>20</td>
      <td>41514559</td>
      <td>A</td>
      <td>20:41514559_A_G</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>457657</th>
      <td>457657</td>
      <td>21:45545974:C:T</td>
      <td>T</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>45545974</td>
      <td>C</td>
      <td>21:45545974_C_T</td>
    </tr>
    <tr>
      <th>460945</th>
      <td>460945</td>
      <td>21:46067181:G:C</td>
      <td>C</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46067181</td>
      <td>G</td>
      <td>21:46067181_G_C</td>
    </tr>
    <tr>
      <th>466362</th>
      <td>466362</td>
      <td>21:46876064:G:C</td>
      <td>C</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>46876064</td>
      <td>G</td>
      <td>21:46876064_G_C</td>
    </tr>
    <tr>
      <th>471833</th>
      <td>471833</td>
      <td>21:47773893:C:G</td>
      <td>G</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47773893</td>
      <td>C</td>
      <td>21:47773893_C_G</td>
    </tr>
    <tr>
      <th>472250</th>
      <td>472250</td>
      <td>21:47831648:C:T</td>
      <td>T</td>
      <td>0.000000e+00</td>
      <td>0.000000</td>
      <td>True</td>
      <td>361194</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>21</td>
      <td>47831648</td>
      <td>C</td>
      <td>21:47831648_C_T</td>
    </tr>
  </tbody>
</table>
<p>473916 rows × 17 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="k">assert</span> <span class="nb">sorted</span><span class="p">(</span><span class="n">test</span><span class="p">[</span><span class="s2">&quot;chr&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">unique</span><span class="p">())</span> <span class="o">==</span> <span class="p">[</span><span class="s1">&#39;20&#39;</span><span class="p">,</span> <span class="s1">&#39;21&#39;</span><span class="p">]</span>
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

</div>
 

