---

title: Title


keywords: fastai
sidebar: home_sidebar



nb_path: "notebooks/package/simulation.ipynb"
---
<!--

#################################################
### THIS FILE WAS AUTOGENERATED! DO NOT EDIT! ###
#################################################
# file to edit: notebooks/package/simulation.ipynb
# command to build the docs after a change: nbdev_build_docs

-->

<div class="container" id="notebook-container">
        
    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">

<div class="output_subarea output_stream output_stderr output_text">
<pre>/lab/corradin_biobank/FOR_AN/combinatorial_GWAS/.venv/lib/python3.8/site-packages/kedro/extras/datasets/pandas/csv_dataset.py:158: ParserWarning: Falling back to the &#39;python&#39; engine because the &#39;c&#39; engine does not support regex separators (separators &gt; 1 char and different from &#39;\s+&#39; are interpreted as regex); you can avoid this warning by specifying engine=&#39;python&#39;.
  return pd.read_csv(fs_file, **self._load_args)
WARNING:root:Loading ICD phenotype matrix, this might take a while
WARNING:root:Finished loading ICD10 matrix
</pre>
</div>
</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">from</span> <span class="nn">combinatorial_gwas.high_level</span> <span class="kn">import</span> <span class="n">chromosome_datasource</span><span class="p">,</span> <span class="n">snp_filter</span> 
</pre></div>

    </div>
</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h2 id="SNPInfoUnit" class="doc_header"><code>class</code> <code>SNPInfoUnit</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L25" class="source_link" style="float:right">[source]</a></h2><blockquote><p><code>SNPInfoUnit</code>(<strong><code>negation</code></strong>:<code>str</code>, <strong><code>snp_id</code></strong>:<code>str</code>, <strong><code>geno</code></strong>:<code>str</code>)</p>
</blockquote>
<p>SNPInfoUnit(negation: str, snp_id: str, geno: str)</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h2 id="SimulatedPheno" class="doc_header"><code>class</code> <code>SimulatedPheno</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L32" class="source_link" style="float:right">[source]</a></h2><blockquote><p><code>SimulatedPheno</code>(<strong><code>snps</code></strong>:<a href="/combinatorial_GWAS/simulation.html#SNPInfoUnit"><code>SNPInfoUnit</code></a>, <strong><code>op</code></strong>:<code>str</code>, <strong><code>query</code></strong>:<code>str</code>, <strong><code>pheno_col</code></strong>:<code>array</code>, <strong><code>case_count</code></strong>:<code>int</code>, <strong><code>control_count</code></strong>:<code>int</code>)</p>
</blockquote>
<p>SimulatedPheno(snps: combinatorial_gwas.simulation.SNPInfoUnit, op: str, query: str, pheno_col: <built-in function array>, case_count: int, control_count: int)</p>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_chrom_to_variants_dict" class="doc_header"><code>get_chrom_to_variants_dict</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L43" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_chrom_to_variants_dict</code>(<strong><code>pairs_df</code></strong>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_simulation_geno_df" class="doc_header"><code>get_simulation_geno_df</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L53" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_simulation_geno_df</code>(<strong><code>datasource</code></strong>, <strong><code>chrom_to_variants_dict</code></strong>, <strong><code>sample_ids</code></strong>=<em><code>None</code></em>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_genotype_dict" class="doc_header"><code>get_genotype_dict</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L70" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_genotype_dict</code>(<strong><code>geno_df</code></strong>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

<div class="output_wrapper">
<div class="output">

<div class="output_area">


<div class="output_markdown rendered_html output_subarea ">
<h4 id="get_two_snps_queries" class="doc_header"><code>get_two_snps_queries</code><a href="https://github.com/hoangthienan95/combinatorial_GWAS/tree/master/combinatorial_gwas/simulation.py#L75" class="source_link" style="float:right">[source]</a></h4><blockquote><p><code>get_two_snps_queries</code>(<strong><code>geno_dict</code></strong>:<code>dict</code>, <strong><code>two_snp_rsid_list</code></strong>:<code>List</code>[<code>str</code>], <strong><code>geno_df</code></strong>)</p>
</blockquote>

</div>

</div>

</div>
</div>

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">

</div>
    {% endraw %}

    {% raw %}
    
<div class="cell border-box-sizing code_cell rendered">
<div class="input">

<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="n">queries_dict</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">

<div class="output_area">



<div class="output_text output_subarea output_execute_result">
<pre>{&#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=404360, control_count=83049),
 &#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=274854, control_count=212555),
 &#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=254551, control_count=232858),
 &#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=456027, control_count=31382),
 &#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=58904, control_count=428505),
 &#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=188410, control_count=298999),
 &#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=208713, control_count=278696),
 &#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=7237, control_count=480172),
 &#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=484476, control_count=2933),
 &#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=477317, control_count=10092),
 &#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=476659, control_count=10750),
 &#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=487039, control_count=370),
 &#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=466197, control_count=21212),
 &#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=473356, control_count=14053),
 &#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=474014, control_count=13395),
 &#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=463634, control_count=23775),
 &#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=280567, control_count=206842),
 &#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=190749, control_count=296660),
 &#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=176721, control_count=310688),
 &#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=316442, control_count=170967),
 &#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=40926, control_count=446483),
 &#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=130744, control_count=356665),
 &#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=144772, control_count=342637),
 &#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=5051, control_count=482358),
 &#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=466498, control_count=20911),
 &#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=419651, control_count=67758),
 &#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         1
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=412718, control_count=74691),
 &#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=484853, control_count=2556),
 &#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=342404, control_count=145005),
 &#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         1
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=389251, control_count=98158),
 &#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=396184, control_count=91225),
 &#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=324049, control_count=163360),
 &#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=167865, control_count=319544),
 &#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         0
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=113008, control_count=374401),
 &#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=105786, control_count=381623),
 &#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=188912, control_count=298497),
 &#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=23992, control_count=463417),
 &#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=78849, control_count=408560),
 &#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         0
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=86071, control_count=401338),
 &#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=2945, control_count=484464),
 &#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=449564, control_count=37845),
 &#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=367756, control_count=119653),
 &#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=354017, control_count=133392),
 &#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=482747, control_count=4662),
 &#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=229702, control_count=257707),
 &#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=311510, control_count=175899),
 &#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=325249, control_count=162160),
 &#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=196519, control_count=290890),
 &#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=423924, control_count=63485),
 &#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=288110, control_count=199299),
 &#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=266780, control_count=220629),
 &#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=478025, control_count=9384),
 &#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=61689, control_count=425720),
 &#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=197503, control_count=289906),
 &#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=218833, control_count=268576),
 &#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=7588, control_count=479821),
 &#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=487261, control_count=148),
 &#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=486410, control_count=999),
 &#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=486779, control_count=630),
 &#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=487390, control_count=19),
 &#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=485761, control_count=1648),
 &#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=486612, control_count=797),
 &#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=486243, control_count=1166),
 &#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; (not (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=485632, control_count=1777),
 &#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=21212, control_count=466197),
 &#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=14053, control_count=473356),
 &#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=13395, control_count=474014),
 &#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=23775, control_count=463634),
 &#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=2933, control_count=484476),
 &#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=10092, control_count=477317),
 &#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=10750, control_count=476659),
 &#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=370, control_count=487039),
 &#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=428505, control_count=58904),
 &#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=298999, control_count=188410),
 &#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=278696, control_count=208713),
 &#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=480172, control_count=7237),
 &#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=83049, control_count=404360),
 &#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=212555, control_count=274854),
 &#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=232858, control_count=254551),
 &#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=0.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 0.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=31382, control_count=456027),
 &#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=145005, control_count=342404),
 &#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         0
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=98158, control_count=389251),
 &#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=91225, control_count=396184),
 &#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=163360, control_count=324049),
 &#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=20911, control_count=466498),
 &#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=67758, control_count=419651),
 &#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         0
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=74691, control_count=412718),
 &#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=2556, control_count=484853),
 &#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=446483, control_count=40926),
 &#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=356665, control_count=130744),
 &#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=342637, control_count=144772),
 &#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=482358, control_count=5051),
 &#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=206842, control_count=280567),
 &#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=296660, control_count=190749),
 &#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=310688, control_count=176721),
 &#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=1.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 1.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         1
 2         1
 3         0
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=170967, control_count=316442),
 &#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=257707, control_count=229702),
 &#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=175899, control_count=311510),
 &#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=162160, control_count=325249),
 &#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=290890, control_count=196519),
 &#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=37845, control_count=449564),
 &#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=119653, control_count=367756),
 &#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=133392, control_count=354017),
 &#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=4662, control_count=482747),
 &#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=463417, control_count=23992),
 &#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=408560, control_count=78849),
 &#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         1
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=401338, control_count=86071),
 &#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=484464, control_count=2945),
 &#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=319544, control_count=167865),
 &#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         1
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=374401, control_count=113008),
 &#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=381623, control_count=105786),
 &#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=2.0), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A` == 2.0)) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         1
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=298497, control_count=188912),
 &#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=1648, control_count=485761),
 &#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=797, control_count=486612),
 &#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=1166, control_count=486243),
 &#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=1777, control_count=485632),
 &#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=148, control_count=487261),
 &#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=999, control_count=486410),
 &#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=630, control_count=486779),
 &#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) and ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=19, control_count=487390),
 &#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         0
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    1
 487406    1
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=425720, control_count=61689),
 &#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         1
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    1
 487406    1
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=289906, control_count=197503),
 &#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         1
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    1
 487406    0
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=268576, control_count=218833),
 &#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;not&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or (not (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         1
 1         1
 2         1
 3         1
 4         1
          ..
 487404    1
 487405    0
 487406    1
 487407    1
 487408    1
 Length: 487409, dtype: int64, case_count=479821, control_count=7588),
 &#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 2.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=2.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 2.0))&#39;, pheno_col=0         1
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    0
 487406    0
 487407    0
 487408    1
 Length: 487409, dtype: int64, case_count=63485, control_count=423924),
 &#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 0.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=0.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 0.0))&#39;, pheno_col=0         0
 1         1
 2         0
 3         0
 4         0
          ..
 487404    1
 487405    0
 487406    0
 487407    1
 487408    0
 Length: 487409, dtype: int64, case_count=199299, control_count=288110),
 &#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 1.0))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=1.0)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C` == 1.0))&#39;, pheno_col=0         0
 1         0
 2         1
 3         1
 4         1
          ..
 487404    0
 487405    0
 487406    1
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=220629, control_count=266780),
 &#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C`.isna()))&#39;: SimulatedPheno(snps=[SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:134911816_G_A&#39;, geno=nan), SNPInfoUnit(negation=&#39;&#39;, snp_id=&#39;6:26118570_T_C&#39;, geno=nan)], op=[], query=&#39; ( (`6:134911816_G_A`.isna())) or ( (`6:26118570_T_C`.isna()))&#39;, pheno_col=0         0
 1         0
 2         0
 3         0
 4         0
          ..
 487404    0
 487405    1
 487406    0
 487407    0
 487408    0
 Length: 487409, dtype: int64, case_count=9384, control_count=478025)}</pre>
</div>

</div>

</div>
</div>

</div>
    {% endraw %}

</div>
 

